// krds-react/dist/index.esm.js
import { jsx as e, jsxs as h, Fragment as Ee } from "react/jsx-runtime";
import Re, { useState as ee, createContext as Ve, useContext as Oe, forwardRef as D, useId as Me, useCallback as F, useEffect as J, useRef as j, useMemo as le, useLayoutEffect as It, useImperativeHandle as Gn, isValidElement as Ke, cloneElement as yt, Children as Xn } from "react";
import { createPortal as Jn } from "react-dom";
function m(...t) {
  const a = [];
  for (const n of t)
    if (n) {
      if (typeof n == "string" || typeof n == "number")
        a.push(String(n));
      else if (typeof n == "object" && !Array.isArray(n))
        for (const [s, l] of Object.entries(n))
          l && a.push(s);
      else if (Array.isArray(n)) {
        const s = m(...n);
        s && a.push(s);
      }
    }
  return a.join(" ");
}
function ne({
  value: t,
  defaultValue: a,
  finalValue: n,
  onChange: s = () => {
  }
}) {
  const [l, o] = ee(
    a !== void 0 ? a : n
  ), r = (c, ...d) => {
    o(c), s?.(c, ...d);
  };
  return t !== void 0 ? [t, s, !0] : [l, r, !1];
}
const Pt = Ve(null), At = () => {
  const t = Oe(Pt);
  if (!t)
    throw new Error("Accordion 하위 컴포넌트는 Accordion 컴포넌트 내부에서 사용되어야 합니다.");
  return t;
}, Bt = Ve(null), $t = () => {
  const t = Oe(Bt);
  if (!t)
    throw new Error(
      "AccordionHeader와 AccordionPanel은 AccordionItem 컴포넌트 내부에서 사용되어야 합니다."
    );
  return t;
}, Rt = D(
  ({ value: t, children: a, className: n = "", ...s }, l) => {
    const { expandedValues: o } = At(), r = o.includes(t), c = Me(), d = m("accordion-item", r && "active", n), u = {
      value: t,
      isExpanded: r,
      itemId: c
    };
    return /* @__PURE__ */ e(Bt.Provider, { value: u, children: /* @__PURE__ */ e("div", { className: d, ref: l, ...s, children: a }) });
  }
);
Rt.displayName = "AccordionItem";
const Ot = D(
  ({ children: t, onClick: a, className: n = "", ...s }, l) => {
    const { onToggle: o } = At(), { value: r, isExpanded: c, itemId: d } = $t(), u = `accordion-header-${d}`, g = `accordion-panel-${d}`, v = m("btn-accordion", c && "active"), y = () => {
      o(r), a?.();
    }, x = (k) => {
      (k.key === "Enter" || k.key === " ") && (k.preventDefault(), y());
    };
    return /* @__PURE__ */ e("h5", { className: m("accordion-header", n), ref: l, ...s, children: /* @__PURE__ */ e(
      "button",
      {
        type: "button",
        id: u,
        className: v,
        "aria-expanded": c,
        "aria-controls": g,
        onClick: y,
        onKeyDown: x,
        children: t
      }
    ) });
  }
);
Ot.displayName = "AccordionHeader";
const Ht = D(
  ({ children: t, className: a = "", ...n }, s) => {
    const { itemId: l } = $t(), o = `accordion-header-${l}`, r = `accordion-panel-${l}`;
    return /* @__PURE__ */ e(
      "div",
      {
        id: r,
        "aria-labelledby": o,
        className: m("accordion-collapse", a),
        ref: s,
        ...n,
        children: /* @__PURE__ */ e("div", { className: "accordion-body", children: t })
      }
    );
  }
);
Ht.displayName = "AccordionPanel";
const ft = D(
  ({
    variant: t = "default",
    allowMultiple: a = !0,
    value: n,
    onChange: s,
    defaultValue: l = [],
    children: o,
    className: r = "",
    ...c
  }, d) => {
    const [u, g] = ne({
      value: n,
      defaultValue: l,
      finalValue: [],
      onChange: s
    }), v = F(
      (k) => {
        const E = u.includes(k);
        if (a) {
          const N = E ? u.filter((L) => L !== k) : [...u, k];
          g(N);
        } else
          g(E ? [] : [k]);
      },
      [u, a, g]
    ), y = m(
      "krds-accordion",
      t === "line" && "type-line",
      r
    ), x = {
      expandedValues: u,
      onToggle: v,
      variant: t
    };
    return /* @__PURE__ */ e(Pt.Provider, { value: x, children: /* @__PURE__ */ e("div", { className: y, ref: d, ...c, children: o }) });
  }
);
ft.displayName = "AccordionRoot";
const Qn = Object.assign(ft, {
  Root: ft,
  Item: Rt,
  Header: Ot,
  Panel: Ht
});
Qn.displayName = "Accordion";
const Zn = D(
  ({
    variant: t = "outline",
    color: a = "primary",
    size: n = "medium",
    rounded: s = !1,
    children: l,
    className: o,
    ...r
  }, c) => {
    const d = t === "outline" ? `outline-${a}` : t === "filled" ? `bg-${a}` : `bg-light-${a}`;
    return /* @__PURE__ */ e(
      "span",
      {
        ref: c,
        className: m(
          "krds-badge",
          n,
          d,
          s && "number",
          o
        ),
        ...r,
        children: l
      }
    );
  }
);
Zn.displayName = "Badge";
const ea = D(
  ({ children: t, className: a, ...n }, s) => /* @__PURE__ */ e("div", { ref: s, className: m("krds-badge-wrap", a), ...n, children: t })
);
ea.displayName = "BadgeWrapper";
const ta = Re.forwardRef(
  ({ items: t, className: a, ariaLabel: n = "현재 경로", ...s }, l) => /* @__PURE__ */ e(
    "nav",
    {
      ref: l,
      className: m("krds-breadcrumb-wrap", a),
      "aria-label": n,
      ...s,
      children: /* @__PURE__ */ e("ol", { className: "breadcrumb", role: "list", children: t.map((o, r) => {
        const c = r === t.length - 1, d = m(r === 0 && "home");
        return /* @__PURE__ */ e("li", { className: d || void 0, children: /* @__PURE__ */ e(
          "a",
          {
            href: o.disabled ? void 0 : o.href,
            className: m("txt"),
            onClick: o.disabled ? (u) => u.preventDefault() : void 0,
            "aria-current": c ? "page" : void 0,
            "aria-disabled": o.disabled ? "true" : void 0,
            children: o.text
          }
        ) }, r);
      }) })
    }
  )
);
ta.displayName = "Breadcrumb";
const na = ({
  as: t,
  children: a,
  variant: n = "primary",
  size: s = "large",
  className: l = "",
  role: o = "button",
  disabled: r = !1,
  ...c
}, d) => {
  const u = t || "button", g = c.type || (u === "button" ? "button" : void 0), v = m("krds-btn", n, s, r && "disabled", l);
  return /* @__PURE__ */ e(
    u,
    {
      ref: d,
      className: v,
      disabled: r,
      "aria-disabled": u !== "button" ? r : void 0,
      onKeyDown: (x) => {
        (x.key === "Enter" || x.key === " ") && (x.preventDefault(), r || x.currentTarget.click());
      },
      role: o,
      type: g,
      ...c,
      children: a
    }
  );
}, Vt = D(na);
Vt.displayName = "Button";
const re = Vt, aa = D(
  ({
    mode: t = "single",
    position: a = "bottom",
    value: n,
    defaultValue: s = "",
    startDate: l,
    endDate: o,
    defaultStartDate: r = "",
    defaultEndDate: c = "",
    disabledDates: d = [],
    eventDates: u = [],
    isOpen: g,
    defaultIsOpen: v = !1,
    onOpenChange: y,
    onChange: x,
    onRangeChange: k,
    onYearChange: E,
    onMonthChange: N,
    onTodayClick: L,
    onConfirm: O,
    onCancel: R,
    label: M = "기간선택",
    inputId: S = "cal",
    placeholder: b = "YYYY.MM.DD",
    startPlaceholder: T = "시작날짜",
    endPlaceholder: B = "종료날짜",
    startTitle: Y = "시작날짜 입력",
    endTitle: V = "종료날짜 입력",
    openButtonLabel: U = "달력 열기",
    prevButtonLabel: G = "이전 달",
    nextButtonLabel: K = "다음 달",
    yearSelectLabel: H = "연도 선택",
    monthSelectLabel: _ = "월 선택",
    todayButtonText: te = "오늘",
    cancelButtonText: ie = "취소",
    confirmButtonText: ce = "확인",
    disabled: oe = !1,
    readOnly: $ = !1,
    className: Q,
    ...me
  }, de) => {
    const [ge, Ne] = ne({
      value: g,
      defaultValue: v,
      finalValue: !1,
      onChange: y
    }), [ke, be] = ne({
      value: n,
      defaultValue: s,
      finalValue: "",
      onChange: x
    }), [X, ue] = ne({
      value: l,
      defaultValue: r,
      finalValue: "",
      onChange: (C) => {
        k && k(C, Le);
      }
    }), [Le, Se] = ne({
      value: o,
      defaultValue: c,
      finalValue: "",
      onChange: (C) => {
        k && k(X, C);
      }
    }), we = F((C) => {
      if (!C) return null;
      const P = C.split(".");
      if (P.length !== 3) return null;
      const i = parseInt(P[0]), p = parseInt(P[1]), f = parseInt(P[2]);
      return isNaN(i) || isNaN(p) || isNaN(f) || p < 1 || p > 12 || f < 1 || f > 31 ? null : new Date(i, p - 1, f);
    }, []), Fe = (C) => `${C.getFullYear()}.${(C.getMonth() + 1).toString().padStart(2, "0")}.${C.getDate().toString().padStart(2, "0")}`, _e = F(() => {
      const C = /* @__PURE__ */ new Date();
      if (t === "single" && ke) {
        const P = we(ke);
        if (P)
          return {
            year: P.getFullYear(),
            month: P.getMonth() + 1
          };
      } else if (t === "range" && X) {
        const P = we(X);
        if (P)
          return {
            year: P.getFullYear(),
            month: P.getMonth() + 1
          };
      }
      return {
        year: C.getFullYear(),
        month: C.getMonth() + 1
      };
    }, [t, ke, X, we]), [ae, Ce] = ee(_e);
    J(() => {
      const C = _e();
      Ce(C);
    }, [ke, X, t, _e]);
    const [We, xe] = ee(!1), [Ie, Ae] = ee(!1), [je, Ge] = ee(0), [He, ze] = ee(null), Xe = j(null), Je = le(() => {
      const C = (/* @__PURE__ */ new Date()).getFullYear();
      return Array.from({ length: 101 }, (P, i) => {
        const p = C - 50 + i;
        return {
          value: p,
          label: `${p}년`,
          isActive: p === ae.year,
          isDisabled: !1
        };
      });
    }, [ae.year]), at = le(() => Array.from({ length: 12 }, (C, P) => {
      const i = P + 1;
      return {
        value: i,
        label: `${i.toString().padStart(2, "0")}월`,
        isActive: i === ae.month,
        isDisabled: !1
      };
    }), [ae.month]), Qe = le(() => {
      const { year: C, month: P } = ae, i = /* @__PURE__ */ new Date(), p = new Date(C, P - 1, 1), f = new Date(C, P, 0), w = p.getDay(), A = f.getDate(), W = [], z = P === 1 ? 12 : P - 1, fe = P === 1 ? C - 1 : C, Z = new Date(fe, z, 0).getDate();
      for (let ve = w - 1; ve >= 0; ve--) {
        const tt = Z - ve, ut = `${fe}.${z.toString().padStart(2, "0")}.${tt.toString().padStart(2, "0")}`;
        W.push({
          year: fe,
          month: z,
          day: tt,
          isOld: !0,
          isDayOff: (w - 1 - ve) % 7 === 0,
          // Sunday
          isDisabled: d.includes(ut),
          hasEvent: u.includes(ut)
        });
      }
      for (let ve = 1; ve <= A; ve++) {
        const ut = new Date(C, P - 1, ve).getDay(), Et = `${C}.${P.toString().padStart(2, "0")}.${ve.toString().padStart(2, "0")}`, jn = C === i.getFullYear() && P === i.getMonth() + 1 && ve === i.getDate();
        W.push({
          year: C,
          month: P,
          day: ve,
          isToday: jn,
          isDayOff: ut === 0,
          // Sunday
          isDisabled: d.includes(Et),
          hasEvent: u.includes(Et)
        });
      }
      const Ue = 42 - W.length, et = P === 12 ? 1 : P + 1, Tt = P === 12 ? C + 1 : C;
      for (let ve = 1; ve <= Ue; ve++) {
        const tt = `${Tt}.${et.toString().padStart(2, "0")}.${ve.toString().padStart(2, "0")}`;
        W.push({
          year: Tt,
          month: et,
          day: ve,
          isNew: !0,
          isDayOff: W.length % 7 === 0,
          isDisabled: d.includes(tt),
          hasEvent: u.includes(tt)
        });
      }
      return W;
    }, [ae, d, u]), st = () => {
      oe || $ || (Ne(!0), setTimeout(() => {
        Xe.current?.focus();
      }, 50));
    }, Pe = F(() => {
      Ne(!1), xe(!1), Ae(!1);
    }, [Ne]), Ye = () => {
      const C = ae.month === 1 ? 12 : ae.month - 1, P = ae.month === 1 ? ae.year - 1 : ae.year;
      Ce({ year: P, month: C }), E?.(P), N?.(C);
    }, Ze = () => {
      const C = ae.month === 12 ? 1 : ae.month + 1, P = ae.month === 12 ? ae.year + 1 : ae.year;
      Ce({ year: P, month: C }), E?.(P), N?.(C);
    }, rt = (C) => {
      C.isDisabled || (Ce((P) => ({ ...P, year: C.value })), xe(!1), E?.(C.value));
    }, lt = (C) => {
      C.isDisabled || (Ce((P) => ({ ...P, month: C.value })), Ae(!1), N?.(C.value));
    }, I = (C) => {
      if (C.isDisabled || C.isOld || C.isNew) return;
      const P = `${C.year}.${C.month.toString().padStart(2, "0")}.${C.day.toString().padStart(2, "0")}`;
      if (t === "single")
        be(P);
      else {
        const i = new Date(C.year, C.month - 1, C.day);
        if (He) {
          const p = new Date(He.year, He.month - 1, He.day);
          if (i < p) {
            ue(P), Se(""), ze(C);
            return;
          }
        }
        Ge(je + 1), je % 2 === 0 ? (ue(P), Se(""), ze(C)) : (Se(P), ze(null));
      }
    }, q = () => {
      const C = /* @__PURE__ */ new Date(), P = Fe(C);
      t === "single" ? be(P) : (ue(P), Se(P)), Ce({
        year: C.getFullYear(),
        month: C.getMonth() + 1
      }), Ge(0), ze(null), L?.();
    }, he = () => {
      Pe(), O?.();
    }, pe = () => {
      Pe(), R?.();
    }, De = (C) => {
      be(C);
      const P = we(C);
      P && Ce({
        year: P.getFullYear(),
        month: P.getMonth() + 1
      });
    }, se = (C) => {
      ue(C);
      const P = we(C);
      P && Ce({
        year: P.getFullYear(),
        month: P.getMonth() + 1
      });
    }, Be = (C) => {
      Se(C);
      const P = we(C);
      P && Ce({
        year: P.getFullYear(),
        month: P.getMonth() + 1
      });
    }, Te = (C) => {
      if (t === "single") {
        const P = ke === `${C.year}.${C.month.toString().padStart(2, "0")}.${C.day.toString().padStart(2, "0")}`;
        return { isPeriod: P, isStart: P, isEnd: P };
      } else {
        const P = `${C.year}.${C.month.toString().padStart(2, "0")}.${C.day.toString().padStart(2, "0")}`, i = X === P, p = Le === P;
        if (!Le)
          return { isPeriod: i, isStart: i, isEnd: p };
        const f = new Date(C.year, C.month - 1, C.day), w = X.split(".").map(Number), A = Le.split(".").map(Number), W = new Date(w[0], w[1] - 1, w[2]), z = new Date(A[0], A[1] - 1, A[2]);
        return { isPeriod: f >= W && f <= z, isStart: i, isEnd: p };
      }
    };
    return J(() => {
      const C = (i) => {
        if (!ge) return;
        i.target.closest(".calendar-conts") || Pe();
      }, P = (i) => {
        i.key === "Escape" && ge && Pe();
      };
      return document.addEventListener("keydown", P), document.addEventListener("click", C), () => {
        document.removeEventListener("keydown", P), document.removeEventListener("click", C);
      };
    }, [ge]), /* @__PURE__ */ h("div", { className: "form-group", ref: de, ...me, children: [
      /* @__PURE__ */ e("div", { className: "form-tit", children: /* @__PURE__ */ e("label", { htmlFor: S, children: M }) }),
      /* @__PURE__ */ h("div", { className: "form-conts calendar-conts", children: [
        /* @__PURE__ */ h("div", { className: "calendar-input", children: [
          t === "single" ? /* @__PURE__ */ e(
            "input",
            {
              type: "text",
              className: "krds-input datepicker cal",
              placeholder: b,
              id: S,
              value: ke,
              onChange: (C) => De(C.target.value),
              disabled: oe,
              readOnly: $
            }
          ) : /* @__PURE__ */ h("ul", { className: "input-group range set", children: [
            /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e(
              "input",
              {
                type: "text",
                className: "krds-input datepicker",
                placeholder: T,
                title: Y,
                value: X,
                onChange: (C) => se(C.target.value),
                disabled: oe,
                readOnly: $
              }
            ) }),
            /* @__PURE__ */ e("li", { className: "mark", children: "-" }),
            /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e(
              "input",
              {
                type: "text",
                className: "krds-input datepicker",
                placeholder: B,
                title: V,
                value: Le,
                onChange: (C) => Be(C.target.value),
                disabled: oe,
                readOnly: $
              }
            ) })
          ] }),
          /* @__PURE__ */ h(
            "button",
            {
              type: "button",
              className: "krds-btn medium icon form-btn-datepicker",
              onClick: st,
              "aria-expanded": ge,
              disabled: oe,
              children: [
                /* @__PURE__ */ e("span", { className: "sr-only", children: U }),
                /* @__PURE__ */ e("i", { className: "svg-icon ico-calendar" })
              ]
            }
          )
        ] }),
        /* @__PURE__ */ e("div", { className: m("krds-calendar-area", { active: ge }), children: /* @__PURE__ */ h(
          "div",
          {
            className: m("calendar-wrap", a, { single: t === "single" }),
            "aria-label": "달력",
            tabIndex: 0,
            ref: Xe,
            children: [
              /* @__PURE__ */ h("div", { className: "calendar-head", children: [
                /* @__PURE__ */ e("button", { type: "button", className: "btn-cal-move prev", onClick: Ye, children: /* @__PURE__ */ e("span", { className: "sr-only", children: G }) }),
                /* @__PURE__ */ h("div", { className: "calendar-switch-wrap", children: [
                  /* @__PURE__ */ h("div", { className: "calendar-drop-down", children: [
                    /* @__PURE__ */ h(
                      "button",
                      {
                        type: "button",
                        className: "btn-cal-switch year",
                        "aria-label": H,
                        onClick: () => {
                          xe(!We), Ae(!1);
                        },
                        "aria-expanded": We,
                        children: [
                          ae.year,
                          "년"
                        ]
                      }
                    ),
                    /* @__PURE__ */ e(
                      "div",
                      {
                        className: m("calendar-select calendar-year-wrap", {
                          active: We
                        }),
                        children: /* @__PURE__ */ e("ul", { className: "sel year", role: "listbox", children: Je.map((C) => /* @__PURE__ */ e("li", { role: "none", children: /* @__PURE__ */ e(
                          "button",
                          {
                            type: "button",
                            role: "option",
                            "aria-selected": C.value === ae.year,
                            className: m({
                              active: C.value === ae.year
                            }),
                            onClick: () => rt(C),
                            disabled: C.isDisabled,
                            children: C.label
                          }
                        ) }, C.value)) })
                      }
                    )
                  ] }),
                  /* @__PURE__ */ h("div", { className: "calendar-drop-down", children: [
                    /* @__PURE__ */ h(
                      "button",
                      {
                        type: "button",
                        className: "btn-cal-switch month",
                        "aria-label": _,
                        onClick: () => {
                          Ae(!Ie), xe(!1);
                        },
                        "aria-expanded": Ie,
                        children: [
                          ae.month.toString().padStart(2, "0"),
                          "월"
                        ]
                      }
                    ),
                    /* @__PURE__ */ e(
                      "div",
                      {
                        className: m("calendar-select calendar-mon-wrap", {
                          active: Ie
                        }),
                        children: /* @__PURE__ */ e("ul", { className: "sel month", role: "listbox", children: at.map((C) => /* @__PURE__ */ e("li", { role: "none", children: /* @__PURE__ */ e(
                          "button",
                          {
                            type: "button",
                            role: "option",
                            "aria-selected": C.value === ae.month,
                            className: m({
                              active: C.value === ae.month
                            }),
                            onClick: () => lt(C),
                            disabled: C.isDisabled,
                            children: C.label
                          }
                        ) }, C.value)) })
                      }
                    )
                  ] })
                ] }),
                /* @__PURE__ */ e("button", { type: "button", className: "btn-cal-move next", onClick: Ze, children: /* @__PURE__ */ e("span", { className: "sr-only", children: K }) })
              ] }),
              /* @__PURE__ */ e("div", { className: "calendar-body", children: /* @__PURE__ */ e("div", { className: "calendar-table-wrap", children: /* @__PURE__ */ h("table", { className: "calendar-tbl", children: [
                /* @__PURE__ */ h("caption", { children: [
                  ae.year,
                  "년 ",
                  ae.month,
                  "월"
                ] }),
                /* @__PURE__ */ e("thead", { children: /* @__PURE__ */ h("tr", { children: [
                  /* @__PURE__ */ e("th", { children: "일" }),
                  /* @__PURE__ */ e("th", { children: "월" }),
                  /* @__PURE__ */ e("th", { children: "화" }),
                  /* @__PURE__ */ e("th", { children: "수" }),
                  /* @__PURE__ */ e("th", { children: "목" }),
                  /* @__PURE__ */ e("th", { children: "금" }),
                  /* @__PURE__ */ e("th", { children: "토" })
                ] }) }),
                /* @__PURE__ */ e("tbody", { children: Array.from({ length: 6 }, (C, P) => /* @__PURE__ */ e("tr", { children: Array.from({ length: 7 }, (i, p) => {
                  const f = P * 7 + p, w = Qe[f];
                  if (!w) return /* @__PURE__ */ e("td", {}, p);
                  const { isPeriod: A, isStart: W, isEnd: z } = Te(w), fe = m({
                    old: w.isOld,
                    new: w.isNew,
                    today: w.isToday,
                    disabled: w.isDisabled,
                    "day-off": w.isDayOff,
                    "day-event": w.hasEvent,
                    period: A,
                    start: W,
                    end: z
                  });
                  return /* @__PURE__ */ e(
                    "td",
                    {
                      className: fe,
                      "data-date": `${w.year}.${w.month.toString().padStart(2, "0")}.${w.day.toString().padStart(2, "0")}`,
                      children: /* @__PURE__ */ e(
                        "button",
                        {
                          type: "button",
                          className: "btn-set-date",
                          onClick: () => I(w),
                          disabled: w.isDisabled || w.isOld || w.isNew,
                          "aria-pressed": A,
                          "aria-label": w.hasEvent ? `${w.day} 일정있음` : w.isToday ? `${w.day} 오늘` : void 0,
                          children: /* @__PURE__ */ e("span", { children: w.day })
                        }
                      )
                    },
                    p
                  );
                }) }, P)) })
              ] }) }) }),
              /* @__PURE__ */ e("div", { className: "calendar-footer", children: /* @__PURE__ */ h("div", { className: "calendar-btn-wrap", children: [
                /* @__PURE__ */ e("button", { type: "button", className: "krds-btn small text", onClick: q, children: te }),
                /* @__PURE__ */ e("button", { type: "button", className: "krds-btn small tertiary", onClick: pe, children: ie }),
                /* @__PURE__ */ e("button", { type: "button", className: "krds-btn small primary", onClick: he, children: ce })
              ] }) })
            ]
          }
        ) })
      ] })
    ] });
  }
);
aa.displayName = "Calendar";
const sa = D(
  ({
    id: t,
    name: a,
    defaultValue: n,
    checked: s,
    disabled: l = !1,
    size: o = "medium",
    children: r,
    description: c,
    onChange: d,
    className: u,
    label: g,
    ...v
  }, y) => {
    const [x, k] = ne({
      value: s,
      defaultValue: n,
      onChange: (L) => {
        d && d({
          target: { checked: L }
        });
      }
    }), E = (L) => {
      k(L.target.checked);
    }, N = m("krds-form-check", o, u);
    return /* @__PURE__ */ h("div", { className: N, children: [
      /* @__PURE__ */ e(
        "input",
        {
          type: "checkbox",
          id: t,
          name: a,
          checked: x,
          disabled: l,
          onChange: E,
          ref: y,
          "aria-label": g || "레이블",
          ...v
        }
      ),
      /* @__PURE__ */ e("label", { htmlFor: t, children: r || g }),
      c && /* @__PURE__ */ e("div", { className: "krds-form-check-cnt", children: /* @__PURE__ */ e("p", { className: "krds-form-check-p", children: c }) })
    ] });
  }
);
sa.displayName = "Checkbox";
const ra = D(
  ({
    id: t,
    name: a,
    defaultValue: n,
    checked: s,
    disabled: l = !1,
    size: o = "medium",
    children: r,
    onChange: c,
    className: d,
    ...u
  }, g) => {
    const [v, y] = ee(!1), x = m("krds-form-chip", o, v && "focus", d), [k, E] = ne({
      value: s,
      defaultValue: n,
      onChange: (R) => {
        c && c({
          target: { checked: R }
        });
      }
    });
    return /* @__PURE__ */ h("div", { className: x, children: [
      /* @__PURE__ */ e(
        "input",
        {
          type: "checkbox",
          className: "checkbox",
          id: t,
          name: a,
          checked: k,
          disabled: l,
          onChange: (R) => {
            E(R.target.checked);
          },
          onFocus: () => {
            y(!0);
          },
          onBlur: () => {
            y(!1);
          },
          ref: g,
          ...u
        }
      ),
      /* @__PURE__ */ e("label", { className: "krds-form-chip-outline", htmlFor: t, children: r })
    ] });
  }
);
ra.displayName = "CheckboxChip";
const la = ({ children: t, column: a = !1, className: n }) => /* @__PURE__ */ e("div", { className: m("krds-check-area", a && "chk-column", n), children: t });
la.displayName = "CheckboxGroup";
const oa = D(
  ({
    title: t = "1단계 : 코치 마크",
    description: a = "1단계 코치 마크 내용입니다.",
    currentStep: n = 1,
    totalSteps: s = 4,
    onSkip: l,
    onPrevious: o,
    onNext: r,
    children: c,
    isVisible: d = !0,
    className: u = "",
    ...g
  }, v) => {
    if (!d)
      return null;
    const y = () => {
      l?.();
    }, x = () => {
      o?.();
    }, k = () => {
      r?.();
    }, E = n > 1;
    return /* @__PURE__ */ h(
      "div",
      {
        ref: v,
        className: m("txt-box bg-white krds-coach-mark", u),
        ...g,
        children: [
          /* @__PURE__ */ h("div", { className: "coach-balloon", children: [
            /* @__PURE__ */ e("h5", { className: "sr-only", children: "따라하기 가이드" }),
            /* @__PURE__ */ e("h6", { className: "coach-tit", children: t }),
            /* @__PURE__ */ e("p", { className: "desc", children: a }),
            /* @__PURE__ */ h("div", { className: "coach-controls", children: [
              /* @__PURE__ */ h("div", { className: "num", children: [
                /* @__PURE__ */ e("span", { className: "sr-only", children: "현재 단계" }),
                /* @__PURE__ */ e("strong", { children: n }),
                /* @__PURE__ */ e("span", { className: "sr-only", children: "총 단계" }),
                /* @__PURE__ */ e("span", { children: s })
              ] }),
              /* @__PURE__ */ h("div", { className: "btn-wrap", children: [
                /* @__PURE__ */ e(re, { variant: "text", size: "small", onClick: y, children: "그만보기" }),
                E && /* @__PURE__ */ e(re, { variant: "text", size: "small", onClick: x, children: "이전으로" }),
                /* @__PURE__ */ e(re, { variant: "tertiary", size: "small", onClick: k, children: "다음으로" })
              ] })
            ] })
          ] }),
          /* @__PURE__ */ e("div", { children: c })
        ]
      }
    );
  }
);
oa.displayName = "CoachMark";
const ia = D(
  ({
    label: t,
    title: a,
    children: n,
    position: s = "top",
    alignment: l = "left",
    open: o,
    defaultOpen: r = !1,
    onOpenChange: c,
    className: d,
    ...u
  }, g) => {
    const v = j(null), y = j(null), x = j(null), k = j(null), [E, N] = ne({
      value: o,
      defaultValue: r,
      onChange: c
    }), L = m("krds-contextual-help", s, l, d), O = () => {
      N(!E);
    }, R = () => {
      N(!1), x.current?.focus();
    }, M = () => {
      if (!E || !y.current || !v.current) return;
      const b = window.innerWidth <= 768, T = v.current.querySelector(".tooltip-action");
      if (b && T) {
        const B = getComputedStyle(document.querySelector(":root")), Y = parseInt(
          B.getPropertyValue("--krds-contents-padding-x")?.replace("px", "") || "20"
        ), U = T.getBoundingClientRect().left - Y, G = document.body.clientWidth - Y * 2;
        y.current.style.left = `-${U}px`, y.current.style.width = `${G}px`;
      } else
        y.current.style.left = "", y.current.style.right = "", y.current.style.width = "";
    }, S = (b) => {
      const T = [
        "button:not([disabled])",
        'input:not([disabled]):not([type="hidden"])',
        "select:not([disabled])",
        "textarea:not([disabled])",
        'a[href]:not([tabindex="-1"])',
        '[tabindex]:not([tabindex="-1"]):not([disabled])',
        '[contenteditable="true"]'
      ].join(", "), B = b.querySelectorAll(T), Y = Array.from(B).filter((V) => {
        const U = V;
        return U.offsetParent !== null && getComputedStyle(U).visibility !== "hidden";
      });
      Y.length > 0 ? Y[0].focus() : y.current?.querySelector(
        ".tooltip-close"
      )?.focus();
    };
    return J(() => {
      const b = (Y) => {
        v.current && !v.current.contains(Y.target) && N(!1);
      }, T = (Y) => {
        Y.key === "Escape" && E && R();
      }, B = () => {
        M();
      };
      if (E)
        return M(), requestAnimationFrame(() => {
          k.current && S(k.current);
        }), document.addEventListener("click", b), document.addEventListener("keydown", T), window.addEventListener("resize", B), () => {
          document.removeEventListener("keydown", T), document.removeEventListener("click", b), window.removeEventListener("resize", B);
        };
    }, [E]), /* @__PURE__ */ h("div", { ref: g || v, className: L, ...u, children: [
      t && /* @__PURE__ */ e("p", { className: "tooltip-txt", children: t }),
      /* @__PURE__ */ h("div", { className: "tooltip-action", children: [
        /* @__PURE__ */ h(
          re,
          {
            ref: x,
            variant: "icon",
            size: "medium",
            className: "tooltip-btn",
            onClick: O,
            "aria-expanded": E,
            children: [
              /* @__PURE__ */ e("span", { className: "sr-only", children: "도움말" }),
              /* @__PURE__ */ e("i", { className: "svg-icon ico-tooltip" })
            ]
          }
        ),
        /* @__PURE__ */ h(
          "div",
          {
            ref: y,
            className: m("tooltip-popover", E && "show"),
            role: "tooltip",
            children: [
              /* @__PURE__ */ e("h4", { className: "tooltip-title", children: a }),
              /* @__PURE__ */ e("div", { ref: k, className: "tooltip-contents", children: n }),
              /* @__PURE__ */ h(
                re,
                {
                  variant: "icon",
                  size: "xsmall",
                  className: "tooltip-close",
                  onClick: R,
                  "aria-label": "닫기",
                  children: [
                    /* @__PURE__ */ e("span", { className: "sr-only", children: "닫기" }),
                    /* @__PURE__ */ e("i", { className: "svg-icon ico-modal-close" })
                  ]
                }
              )
            ]
          }
        )
      ] })
    ] });
  }
);
ia.displayName = "ContextualHelp";
const Ft = D(
  ({ alerts: t, className: a, ...n }, s) => /* @__PURE__ */ e("div", { ref: s, className: m("main-urgent-wrap", a), ...n, children: /* @__PURE__ */ e("ul", { className: "krds-critical-alerts", children: t.map((l, o) => /* @__PURE__ */ e(ca, { ...l }, o)) }) })
);
Ft.displayName = "CriticalAlerts";
const ca = ({
  variant: t,
  message: a,
  href: n,
  linkText: s = "자세히 보기"
}) => {
  const l = {
    danger: "긴급",
    ok: "안전",
    info: "안내"
  }[t];
  return /* @__PURE__ */ e("li", { children: /* @__PURE__ */ h("div", { className: "critical-ban", children: [
    /* @__PURE__ */ e("span", { className: m("critical-badge", t), children: l }),
    /* @__PURE__ */ e("p", { className: "critical-txt", children: a }),
    n && /* @__PURE__ */ h(re, { as: "a", href: n, variant: "link", size: "medium", className: "basic", children: [
      /* @__PURE__ */ e("span", { className: "m-hide", children: s }),
      /* @__PURE__ */ e("i", { className: "svg-icon ico-angle right" })
    ] })
  ] }) });
}, Cs = Ft, da = D(
  ({
    value: t,
    defaultValue: a,
    onChange: n,
    label: s,
    hint: l,
    error: o,
    success: r,
    information: c,
    size: d = "large",
    isCalendarOpen: u,
    defaultIsCalendarOpen: g = !1,
    onCalendarOpenChange: v,
    calendarPosition: y = "bottom",
    disabledDates: x = [],
    eventDates: k = [],
    onYearChange: E,
    onMonthChange: N,
    onTodayClick: L,
    onConfirm: O,
    onCancel: R,
    openButtonLabel: M = "달력 열기",
    prevButtonLabel: S = "이전 달",
    nextButtonLabel: b = "다음 달",
    yearSelectLabel: T = "연도 선택",
    monthSelectLabel: B = "월 선택",
    todayButtonText: Y = "오늘",
    cancelButtonText: V = "취소",
    confirmButtonText: U = "확인",
    disabled: G = !1,
    readOnly: K = !1,
    className: H = "",
    id: _,
    placeholder: te = "YYYY.MM.DD",
    ...ie
  }, ce) => {
    const oe = Me(), $ = _ || oe, [Q, me] = ne({
      value: t,
      defaultValue: a,
      onChange: n,
      finalValue: ""
    }), [de, ge] = ne({
      value: u,
      defaultValue: g,
      finalValue: !1,
      onChange: v
    }), Ne = (I) => {
      if (!I) return null;
      const q = I.split(".");
      if (q.length !== 3) return null;
      const he = parseInt(q[0]), pe = parseInt(q[1]), De = parseInt(q[2]);
      return isNaN(he) || isNaN(pe) || isNaN(De) || pe < 1 || pe > 12 || De < 1 || De > 31 ? null : new Date(he, pe - 1, De);
    }, ke = (I) => `${I.getFullYear()}.${(I.getMonth() + 1).toString().padStart(2, "0")}.${I.getDate().toString().padStart(2, "0")}`, be = () => {
      const I = /* @__PURE__ */ new Date();
      if (Q) {
        const q = Ne(Q);
        if (q)
          return {
            year: q.getFullYear(),
            month: q.getMonth() + 1
          };
      }
      return {
        year: I.getFullYear(),
        month: I.getMonth() + 1
      };
    }, [X, ue] = ee(be);
    J(() => {
      const I = be();
      ue(I);
    }, [Q]);
    const [Le, Se] = ee(!1), [we, Fe] = ee(!1), _e = j(null), ae = le(() => {
      const I = (/* @__PURE__ */ new Date()).getFullYear();
      return Array.from({ length: 101 }, (q, he) => {
        const pe = I - 50 + he;
        return {
          value: pe,
          label: `${pe}년`,
          isActive: pe === X.year,
          isDisabled: !1
        };
      });
    }, [X.year]), Ce = le(() => Array.from({ length: 12 }, (I, q) => {
      const he = q + 1;
      return {
        value: he,
        label: `${he.toString().padStart(2, "0")}월`,
        isActive: he === X.month,
        isDisabled: !1
      };
    }), [X.month]), We = le(() => {
      const { year: I, month: q } = X, he = /* @__PURE__ */ new Date(), pe = new Date(I, q - 1, 1), De = new Date(I, q, 0), se = pe.getDay(), Be = De.getDate(), Te = [], C = q === 1 ? 12 : q - 1, P = q === 1 ? I - 1 : I, i = new Date(P, C, 0).getDate();
      for (let A = se - 1; A >= 0; A--) {
        const W = i - A, z = `${P}.${C.toString().padStart(2, "0")}.${W.toString().padStart(2, "0")}`;
        Te.push({
          year: P,
          month: C,
          day: W,
          isOld: !0,
          isDayOff: (se - 1 - A) % 7 === 0,
          // Sunday
          isDisabled: x.includes(z),
          hasEvent: k.includes(z)
        });
      }
      for (let A = 1; A <= Be; A++) {
        const z = new Date(I, q - 1, A).getDay(), fe = `${I}.${q.toString().padStart(2, "0")}.${A.toString().padStart(2, "0")}`, Z = I === he.getFullYear() && q === he.getMonth() + 1 && A === he.getDate();
        Te.push({
          year: I,
          month: q,
          day: A,
          isToday: Z,
          isDayOff: z === 0,
          // Sunday
          isDisabled: x.includes(fe),
          hasEvent: k.includes(fe)
        });
      }
      const p = 42 - Te.length, f = q === 12 ? 1 : q + 1, w = q === 12 ? I + 1 : I;
      for (let A = 1; A <= p; A++) {
        const W = `${w}.${f.toString().padStart(2, "0")}.${A.toString().padStart(2, "0")}`;
        Te.push({
          year: w,
          month: f,
          day: A,
          isNew: !0,
          isDayOff: Te.length % 7 === 0,
          // Sunday
          isDisabled: x.includes(W),
          hasEvent: k.includes(W)
        });
      }
      return Te;
    }, [X, x, k]), xe = () => {
      G || K || (ge(!0), setTimeout(() => {
        _e.current?.focus();
      }, 50));
    }, Ie = () => {
      ge(!1), Se(!1), Fe(!1);
    }, Ae = () => {
      const I = X.month === 1 ? 12 : X.month - 1, q = X.month === 1 ? X.year - 1 : X.year;
      ue({ year: q, month: I }), E?.(q), N?.(I);
    }, je = () => {
      const I = X.month === 12 ? 1 : X.month + 1, q = X.month === 12 ? X.year + 1 : X.year;
      ue({ year: q, month: I }), E?.(q), N?.(I);
    }, Ge = (I) => {
      I.isDisabled || (ue((q) => ({ ...q, year: I.value })), Se(!1), E?.(I.value));
    }, He = (I) => {
      I.isDisabled || (ue((q) => ({ ...q, month: I.value })), Fe(!1), N?.(I.value));
    }, ze = (I) => {
      if (I.isDisabled || I.isOld || I.isNew) return;
      const q = `${I.year}.${I.month.toString().padStart(2, "0")}.${I.day.toString().padStart(2, "0")}`;
      me(q);
    }, Xe = () => {
      const I = /* @__PURE__ */ new Date(), q = ke(I);
      me(q), ue({
        year: I.getFullYear(),
        month: I.getMonth() + 1
      }), L?.();
    }, Je = () => {
      Ie(), O?.();
    }, at = () => {
      Ie(), R?.();
    }, Qe = (I) => {
      me(I);
      const q = Ne(I);
      q && ue({
        year: q.getFullYear(),
        month: q.getMonth() + 1
      });
    }, st = (I) => {
      const q = `${I.year}.${I.month.toString().padStart(2, "0")}.${I.day.toString().padStart(2, "0")}`;
      return Q === q;
    };
    J(() => {
      const I = (he) => {
        if (!de) return;
        he.target.closest(".calendar-conts") || Ie();
      }, q = (he) => {
        he.key === "Escape" && de && Ie();
      };
      return document.addEventListener("keydown", q), document.addEventListener("click", I), () => {
        document.removeEventListener("keydown", q), document.removeEventListener("click", I);
      };
    }, [de]);
    const Pe = !!o, Ye = !!r && !Pe, Ze = !!c && !Pe && !Ye, rt = m(
      "form-group",
      Pe && "is-error",
      Ye && "is-success",
      Ze && "is-information",
      H
    ), lt = () => Pe ? /* @__PURE__ */ e("p", { className: "form-hint-invalid", children: o }) : Ye ? /* @__PURE__ */ e("p", { className: "form-hint-success", children: r }) : Ze ? /* @__PURE__ */ e("p", { className: "form-hint-information", children: c }) : l ? /* @__PURE__ */ e("p", { className: "form-hint", children: l }) : null;
    return /* @__PURE__ */ h("div", { className: rt, children: [
      s && /* @__PURE__ */ e("div", { className: "form-tit", children: /* @__PURE__ */ e("label", { htmlFor: $, children: s }) }),
      /* @__PURE__ */ e("div", { className: "form-conts", children: /* @__PURE__ */ h("div", { className: "form-conts calendar-conts", children: [
        /* @__PURE__ */ h("div", { className: "calendar-input", children: [
          /* @__PURE__ */ e(
            "input",
            {
              ref: ce,
              type: "text",
              className: m("krds-input datepicker cal", d),
              placeholder: te,
              id: $,
              value: Q,
              onChange: (I) => Qe(I.target.value),
              disabled: G,
              readOnly: K,
              ...ie
            }
          ),
          /* @__PURE__ */ h(
            "button",
            {
              type: "button",
              className: "krds-btn medium icon form-btn-datepicker",
              onClick: xe,
              "aria-expanded": de,
              disabled: G,
              children: [
                /* @__PURE__ */ e("span", { className: "sr-only", children: M }),
                /* @__PURE__ */ e("i", { className: "svg-icon ico-calendar" })
              ]
            }
          )
        ] }),
        /* @__PURE__ */ e("div", { className: m("krds-calendar-area", { active: de }), children: /* @__PURE__ */ h(
          "div",
          {
            className: m("calendar-wrap", y),
            "aria-label": "달력",
            tabIndex: 0,
            ref: _e,
            children: [
              /* @__PURE__ */ h("div", { className: "calendar-head", children: [
                /* @__PURE__ */ e("button", { type: "button", className: "btn-cal-move prev", onClick: Ae, children: /* @__PURE__ */ e("span", { className: "sr-only", children: S }) }),
                /* @__PURE__ */ h("div", { className: "calendar-switch-wrap", children: [
                  /* @__PURE__ */ h("div", { className: "calendar-drop-down", children: [
                    /* @__PURE__ */ h(
                      "button",
                      {
                        type: "button",
                        className: "btn-cal-switch year",
                        "aria-label": T,
                        onClick: () => {
                          Se(!Le), Fe(!1);
                        },
                        "aria-expanded": Le,
                        children: [
                          X.year,
                          "년"
                        ]
                      }
                    ),
                    /* @__PURE__ */ e(
                      "div",
                      {
                        className: m("calendar-select calendar-year-wrap", {
                          active: Le
                        }),
                        children: /* @__PURE__ */ e("ul", { className: "sel year", role: "listbox", children: ae.map((I) => /* @__PURE__ */ e("li", { role: "none", children: /* @__PURE__ */ e(
                          "button",
                          {
                            type: "button",
                            role: "option",
                            "aria-selected": I.value === X.year,
                            className: m({
                              active: I.value === X.year
                            }),
                            onClick: () => Ge(I),
                            disabled: I.isDisabled,
                            children: I.label
                          }
                        ) }, I.value)) })
                      }
                    )
                  ] }),
                  /* @__PURE__ */ h("div", { className: "calendar-drop-down", children: [
                    /* @__PURE__ */ h(
                      "button",
                      {
                        type: "button",
                        className: "btn-cal-switch month",
                        "aria-label": B,
                        onClick: () => {
                          Fe(!we), Se(!1);
                        },
                        "aria-expanded": we,
                        children: [
                          X.month.toString().padStart(2, "0"),
                          "월"
                        ]
                      }
                    ),
                    /* @__PURE__ */ e(
                      "div",
                      {
                        className: m("calendar-select calendar-mon-wrap", {
                          active: we
                        }),
                        children: /* @__PURE__ */ e("ul", { className: "sel month", role: "listbox", children: Ce.map((I) => /* @__PURE__ */ e("li", { role: "none", children: /* @__PURE__ */ e(
                          "button",
                          {
                            type: "button",
                            role: "option",
                            "aria-selected": I.value === X.month,
                            className: m({
                              active: I.value === X.month
                            }),
                            onClick: () => He(I),
                            disabled: I.isDisabled,
                            children: I.label
                          }
                        ) }, I.value)) })
                      }
                    )
                  ] })
                ] }),
                /* @__PURE__ */ e("button", { type: "button", className: "btn-cal-move next", onClick: je, children: /* @__PURE__ */ e("span", { className: "sr-only", children: b }) })
              ] }),
              /* @__PURE__ */ e("div", { className: "calendar-body", children: /* @__PURE__ */ e("div", { className: "calendar-table-wrap", children: /* @__PURE__ */ h("table", { className: "calendar-tbl", children: [
                /* @__PURE__ */ h("caption", { children: [
                  X.year,
                  "년 ",
                  X.month,
                  "월"
                ] }),
                /* @__PURE__ */ e("thead", { children: /* @__PURE__ */ h("tr", { children: [
                  /* @__PURE__ */ e("th", { children: "일" }),
                  /* @__PURE__ */ e("th", { children: "월" }),
                  /* @__PURE__ */ e("th", { children: "화" }),
                  /* @__PURE__ */ e("th", { children: "수" }),
                  /* @__PURE__ */ e("th", { children: "목" }),
                  /* @__PURE__ */ e("th", { children: "금" }),
                  /* @__PURE__ */ e("th", { children: "토" })
                ] }) }),
                /* @__PURE__ */ e("tbody", { children: Array.from({ length: 6 }, (I, q) => /* @__PURE__ */ e("tr", { children: Array.from({ length: 7 }, (he, pe) => {
                  const De = q * 7 + pe, se = We[De];
                  if (!se) return /* @__PURE__ */ e("td", {}, pe);
                  const Be = st(se), Te = m({
                    old: se.isOld,
                    new: se.isNew,
                    today: se.isToday,
                    disabled: se.isDisabled,
                    "day-off": se.isDayOff,
                    "day-event": se.hasEvent,
                    period: Be,
                    start: Be,
                    end: Be
                  });
                  return /* @__PURE__ */ e(
                    "td",
                    {
                      className: Te,
                      "data-date": `${se.year}.${se.month.toString().padStart(2, "0")}.${se.day.toString().padStart(2, "0")}`,
                      children: /* @__PURE__ */ e(
                        "button",
                        {
                          type: "button",
                          className: "btn-set-date",
                          onClick: () => ze(se),
                          disabled: se.isDisabled || se.isOld || se.isNew,
                          "aria-pressed": Be,
                          "aria-label": se.hasEvent ? `${se.day} 일정있음` : se.isToday ? `${se.day} 오늘` : void 0,
                          children: /* @__PURE__ */ e("span", { children: se.day })
                        }
                      )
                    },
                    pe
                  );
                }) }, q)) })
              ] }) }) }),
              /* @__PURE__ */ e("div", { className: "calendar-footer", children: /* @__PURE__ */ h("div", { className: "calendar-btn-wrap", children: [
                /* @__PURE__ */ e(
                  "button",
                  {
                    type: "button",
                    className: "krds-btn small text",
                    onClick: Xe,
                    children: Y
                  }
                ),
                /* @__PURE__ */ e(
                  "button",
                  {
                    type: "button",
                    className: "krds-btn small tertiary",
                    onClick: at,
                    children: V
                  }
                ),
                /* @__PURE__ */ e(
                  "button",
                  {
                    type: "button",
                    className: "krds-btn small primary",
                    onClick: Je,
                    children: U
                  }
                )
              ] }) })
            ]
          }
        ) })
      ] }) }),
      lt()
    ] });
  }
);
da.displayName = "DateInput";
const kt = D(
  ({ buttonText: t, children: a, defaultExpanded: n = !1, expanded: s, onToggle: l, className: o, ...r }, c) => {
    const d = "disclosure-content-" + Me(), [u, g] = ne({
      value: s,
      defaultValue: n,
      finalValue: !1,
      onChange: l
    }), v = () => {
      g(!u);
    };
    return /* @__PURE__ */ h(
      "div",
      {
        ref: c,
        className: m("krds-disclosure", u && "active", o),
        ...r,
        children: [
          /* @__PURE__ */ e(
            "button",
            {
              type: "button",
              className: "btn-conts-expand",
              "aria-expanded": u,
              "aria-controls": d,
              onClick: v,
              children: t
            }
          ),
          /* @__PURE__ */ e("div", { id: d, className: "expand-wrap", children: /* @__PURE__ */ e("div", { className: "expand-in", children: a }) })
        ]
      }
    );
  }
);
kt.displayName = "Disclosure";
const ua = ({ label: t, className: a, children: n }) => {
  const s = /* @__PURE__ */ h("div", { className: m("krds-spinner", a), role: "status", children: [
    /* @__PURE__ */ e("span", { className: "sr-only", children: "로딩 중" }),
    t
  ] });
  return n ? /* @__PURE__ */ h("div", { className: "form-spinner", children: [
    n,
    s
  ] }) : s;
}, ha = D(
  ({
    title: t,
    description: a,
    uploadText: n = "첨부할 파일을 여기에 끌어다 놓거나, 파일 선택 버튼을 눌러 파일을 직접 선택해주세요.",
    maxFiles: s = 10,
    maxFileSize: l = 20 * 1024 * 1024,
    acceptedFileTypes: o,
    files: r,
    onFilesChange: c,
    onFileUpload: d,
    onFileDelete: u,
    onAllFilesDelete: g,
    disabled: v = !1,
    allowDelete: y = !0,
    className: x = "",
    children: k,
    ...E
  }, N) => {
    const [L, O] = ne({
      value: r,
      defaultValue: [],
      onChange: c,
      finalValue: []
    }), [R, M] = ee(!1), S = Me(), b = j(null), T = F(($) => {
      if ($ === 0) return "0B";
      const Q = 1024, me = ["B", "KB", "MB", "GB"], de = Math.floor(Math.log($) / Math.log(Q));
      return parseFloat(($ / Math.pow(Q, de)).toFixed(1)) + me[de];
    }, []), B = F(($) => $.split(".").pop()?.toLowerCase() || "", []), Y = F(
      ($) => {
        if ($.size > l)
          return {
            isValid: !1,
            errorMessage: `등록 가능한 파일 용량을 초과하였습니다.
${T(l)} 미만의 파일만 등록할 수 있습니다.`
          };
        if (o?.length) {
          const Q = B($.name);
          if (!o.includes(Q))
            return {
              isValid: !1,
              errorMessage: `지원하지 않는 파일 형식입니다.
${o.join(", ")} 형식만 업로드 가능합니다.`
            };
        }
        return { isValid: !0 };
      },
      [l, o, T, B]
    ), V = F(
      async ($) => {
        const Q = [], me = s - L.length, de = Math.min($.length, me);
        for (let ge = 0; ge < de; ge++) {
          const Ne = $[ge], ke = Y(Ne), be = {
            id: `${Date.now()}-${ge}`,
            name: Ne.name,
            size: Ne.size,
            type: B(Ne.name),
            status: ke.isValid ? "ready" : "error",
            errorMessage: ke.errorMessage
          };
          if (Q.push(be), ke.isValid && d)
            try {
              be.status = "uploading", await d(Ne), be.status = "completed";
            } catch {
              be.status = "error", be.errorMessage = "파일 업로드에 실패했습니다.";
            }
        }
        Q.length > 0 && O([...L, ...Q]);
      },
      [L, s, Y, B, d, O]
    ), U = F(
      ($) => {
        $?.length && V($);
      },
      [V]
    ), G = F(
      ($) => {
        U($.target.files), $.target.value = "";
      },
      [U]
    ), K = F(
      ($) => {
        $.preventDefault(), v || M(!0);
      },
      [v]
    ), H = F(($) => {
      $.preventDefault(), M(!1);
    }, []), _ = F(
      ($) => {
        $.preventDefault(), M(!1), v || U($.dataTransfer.files);
      },
      [v, U]
    ), te = F(
      ($) => {
        u ? u($) : c?.(L.filter((Q) => Q.id !== $));
      },
      [L, u, c]
    ), ie = F(() => {
      g ? g() : c?.([]);
    }, [g, c]), ce = ($) => {
      switch ($.status) {
        case "uploading":
          return /* @__PURE__ */ h("span", { role: "status", children: [
            /* @__PURE__ */ e(ua, {}),
            /* @__PURE__ */ e("span", { className: "sr-only", children: "업로드 중" })
          ] });
        case "completed":
          return /* @__PURE__ */ e("span", { className: "ico-invalid complete", children: /* @__PURE__ */ e("em", { className: "sr-only", children: "업로드 완료" }) });
        default:
          return null;
      }
    }, oe = ($) => $.status === "uploading" || !y ? null : $.onDownload || $.onPreview ? /* @__PURE__ */ h(Ee, { children: [
      $.onDownload && /* @__PURE__ */ h(re, { variant: "text", size: "medium", onClick: $.onDownload, disabled: v, children: [
        "다운로드 ",
        /* @__PURE__ */ e("i", { className: "svg-icon ico-down" })
      ] }),
      $.onPreview && /* @__PURE__ */ h(re, { variant: "text", size: "medium", onClick: $.onPreview, disabled: v, children: [
        "바로보기 ",
        /* @__PURE__ */ e("i", { className: "svg-icon ico-angle right" })
      ] }),
      $.deletable !== !1 && /* @__PURE__ */ h(
        re,
        {
          variant: "text",
          size: "medium",
          onClick: () => te($.id),
          disabled: v,
          children: [
            "삭제 ",
            /* @__PURE__ */ e("i", { className: "svg-icon ico-delete-fill" })
          ]
        }
      )
    ] }) : $.deletable === !1 ? null : /* @__PURE__ */ h(
      re,
      {
        variant: "text",
        size: "medium",
        onClick: () => te($.id),
        disabled: v,
        children: [
          "삭제 ",
          /* @__PURE__ */ e("i", { className: "svg-icon ico-delete-fill" })
        ]
      }
    );
    return /* @__PURE__ */ h("div", { ref: N, className: m("krds-file-upload line", x), ...E, children: [
      (t || a || k) && /* @__PURE__ */ h("div", { className: "file-head", children: [
        t && /* @__PURE__ */ e("h3", { className: "tit", children: t }),
        a && /* @__PURE__ */ e("div", { children: /* @__PURE__ */ e("p", { children: a }) }),
        k
      ] }),
      /* @__PURE__ */ h(
        "div",
        {
          className: m("file-upload", R && "active"),
          onDragOver: K,
          onDragLeave: H,
          onDrop: _,
          children: [
            /* @__PURE__ */ e("p", { className: "txt", children: n }),
            /* @__PURE__ */ h("div", { className: "file-upload-btn-wrap", children: [
              /* @__PURE__ */ e(
                "input",
                {
                  type: "file",
                  name: "myFile",
                  id: S,
                  hidden: !0,
                  multiple: !0,
                  accept: o?.map(($) => `.${$}`).join(","),
                  onChange: G,
                  disabled: v,
                  ref: b
                }
              ),
              /* @__PURE__ */ e("label", { htmlFor: S, children: /* @__PURE__ */ h(
                re,
                {
                  variant: "primary",
                  size: "medium",
                  disabled: v,
                  onClick: () => {
                    b.current?.click();
                  },
                  children: [
                    /* @__PURE__ */ e("i", { className: "svg-icon ico-upload" }),
                    "파일선택"
                  ]
                }
              ) })
            ] })
          ]
        }
      ),
      L.length > 0 && /* @__PURE__ */ h("div", { className: "file-list", children: [
        /* @__PURE__ */ h("div", { className: "total", children: [
          /* @__PURE__ */ h("span", { className: "current", children: [
            L.length,
            "개"
          ] }),
          " / ",
          s,
          "개"
        ] }),
        /* @__PURE__ */ e("ul", { className: "upload-list", children: L.map(($) => /* @__PURE__ */ h("li", { className: m($.status === "error" && "is-error"), children: [
          /* @__PURE__ */ h(
            "div",
            {
              className: m(
                "file-info",
                ($.onDownload || $.onPreview) && "m-column"
              ),
              children: [
                /* @__PURE__ */ h("div", { className: "file-name", children: [
                  $.name,
                  " [",
                  $.type,
                  ", ",
                  T($.size),
                  "]"
                ] }),
                /* @__PURE__ */ h("div", { className: "btn-wrap", children: [
                  ce($),
                  oe($)
                ] })
              ]
            }
          ),
          $.status === "error" && $.errorMessage && /* @__PURE__ */ e("p", { className: "file-hint-invalid", children: $.errorMessage })
        ] }, $.id)) }),
        L.length > 1 && y && /* @__PURE__ */ e("div", { className: "upload-delete-btn", children: /* @__PURE__ */ h(
          re,
          {
            variant: "tertiary",
            size: "xsmall",
            onClick: ie,
            disabled: v,
            children: [
              "전체 파일 삭제",
              /* @__PURE__ */ e("i", { className: "svg-icon ico-angle right" })
            ]
          }
        ) })
      ] })
    ] });
  }
);
ha.displayName = "FileUpload";
const _t = D(
  ({
    text: t = "이 누리집은 보건복지부 산하기관 누리집입니다.",
    logoAlt: a = "KRDS - Korea Design System",
    className: n,
    ...s
  }, l) => /* @__PURE__ */ h(
    "div",
    {
      ref: l,
      "aria-label": "운영기관 식별자",
      className: m("krds-identifier", n),
      ...s,
      children: [
        /* @__PURE__ */ e("span", { className: "logo", children: /* @__PURE__ */ e("span", { className: "sr-only", children: a }) }),
        /* @__PURE__ */ e("span", { className: "ban-txt", children: t })
      ]
    }
  )
);
_t.displayName = "Identifier";
const ma = ({ quickLinks: t }) => !t || t.length === 0 ? null : /* @__PURE__ */ e("div", { className: "foot-quick", children: /* @__PURE__ */ e("div", { className: "inner", children: t.map((a, n) => /* @__PURE__ */ e(
  "button",
  {
    type: "button",
    className: "link",
    title: `${a.title} 레이어`,
    onClick: a.onClick,
    children: a.title
  },
  n
)) }) }), pa = ({ logo: t }) => {
  const a = t?.src ? /* @__PURE__ */ e(
    "img",
    {
      src: t.src,
      alt: t.alt,
      width: t.width,
      height: t.height,
      style: { width: "100%", height: "100%", objectFit: "contain" }
    }
  ) : /* @__PURE__ */ e("span", { className: "sr-only", children: t?.alt || "KRDS - Korea Design System" }), n = /* @__PURE__ */ e("div", { className: "f-logo", children: a });
  return t?.href ? /* @__PURE__ */ e(
    "a",
    {
      href: t.href,
      target: t.target,
      rel: t.target === "_blank" ? "noopener noreferrer" : void 0,
      children: n
    }
  ) : n;
}, fa = ({
  logo: t,
  address: a,
  contacts: n,
  links: s,
  socialLinks: l,
  defaultLinkTarget: o = "_self"
}) => /* @__PURE__ */ h(Ee, { children: [
  /* @__PURE__ */ e(pa, { logo: t }),
  /* @__PURE__ */ h("div", { className: "f-cnt", children: [
    (a || n) && /* @__PURE__ */ h("div", { className: "f-info", children: [
      a && /* @__PURE__ */ e("p", { className: "info-addr", children: a }),
      n && n.length > 0 && /* @__PURE__ */ e("ul", { className: "info-cs", children: n.map((r, c) => /* @__PURE__ */ h("li", { children: [
        /* @__PURE__ */ e("strong", { className: "strong", children: r.title }),
        /* @__PURE__ */ e("span", { className: "span", children: r.description })
      ] }, c)) })
    ] }),
    (s || l) && /* @__PURE__ */ h("div", { className: "f-link", children: [
      s && s.length > 0 && /* @__PURE__ */ e("div", { className: "link-go", children: s.map((r, c) => {
        const d = r.target || o, u = r.rel || (d === "_blank" ? "noopener noreferrer" : void 0);
        return /* @__PURE__ */ h(
          re,
          {
            as: "a",
            href: r.href,
            target: d,
            rel: u,
            variant: "text",
            size: "medium",
            children: [
              r.text,
              " ",
              /* @__PURE__ */ e("i", { className: "svg-icon ico-angle right" })
            ]
          },
          c
        );
      }) }),
      l && l.length > 0 && /* @__PURE__ */ e("div", { className: "link-sns", children: l.map((r, c) => {
        const d = r.target || o || "_blank", u = r.rel || (d === "_blank" ? "noopener noreferrer" : void 0), g = r.label || ba(r.platform), v = va(r.platform);
        return /* @__PURE__ */ h(
          "a",
          {
            href: r.href,
            target: d,
            rel: u,
            className: "krds-btn xlarge icon border",
            title: d === "_blank" ? "새 창 열기" : void 0,
            children: [
              /* @__PURE__ */ e("span", { className: "sr-only", children: g }),
              /* @__PURE__ */ e("i", { className: `svg-icon ${v}` })
            ]
          },
          c
        );
      }) })
    ] })
  ] })
] }), ga = ({
  bottomLinks: t,
  copyright: a = "© 2023 National Health Insurance Service. All rights reserved.",
  identifierText: n = "이 누리집은 보건복지부 산하기관 누리집입니다.",
  identifier: s,
  defaultLinkTarget: l = "_self",
  hideIdentifier: o = !1
}) => {
  const r = s || { text: n };
  return /* @__PURE__ */ h("div", { className: "f-btm", children: [
    /* @__PURE__ */ h("div", { className: "f-btm-text", children: [
      t && t.length > 0 && /* @__PURE__ */ e("div", { className: "f-menu", children: t.map((c, d) => {
        const u = c.target || l, g = c.rel || (u === "_blank" ? "noopener noreferrer" : void 0);
        return /* @__PURE__ */ e(
          "a",
          {
            href: c.href,
            target: u,
            rel: g,
            className: m(c.isHighlighted && "point"),
            children: c.text
          },
          d
        );
      }) }),
      /* @__PURE__ */ e("p", { className: "f-copy", children: a })
    ] }),
    !o && /* @__PURE__ */ e(
      _t,
      {
        text: r.text,
        logoAlt: r.logo?.alt || "KRDS - Korea Design System"
      }
    )
  ] });
}, ba = (t) => {
  switch (t.toLowerCase()) {
    case "instagram":
      return "인스타그램";
    case "youtube":
      return "유튜브";
    case "x":
      return "X";
    case "twitter":
      return "트위터";
    case "facebook":
      return "페이스북";
    case "blog":
      return "블로그";
    default:
      return t;
  }
}, va = (t) => {
  switch (t.toLowerCase()) {
    case "instagram":
      return "ico-instagram";
    case "youtube":
      return "ico-youtube";
    case "x":
    case "twitter":
      return "ico-sns-x";
    case "facebook":
      return "ico-facebook";
    case "blog":
      return "ico-blog";
    default:
      return "";
  }
}, Na = D(
  ({
    quickLinks: t,
    address: a,
    contacts: n,
    logo: s,
    links: l,
    socialLinks: o,
    bottomLinks: r,
    copyright: c,
    identifierText: d,
    identifier: u,
    defaultLinkTarget: g,
    hideQuickLinks: v = !1,
    hideIdentifier: y = !1,
    className: x,
    ...k
  }, E) => /* @__PURE__ */ h("footer", { ref: E, id: "krds-footer", className: m(x), ...k, children: [
    !v && /* @__PURE__ */ e(ma, { quickLinks: t }),
    /* @__PURE__ */ h("div", { className: "inner", children: [
      /* @__PURE__ */ e(
        fa,
        {
          logo: s,
          address: a,
          contacts: n,
          links: l,
          socialLinks: o,
          defaultLinkTarget: g
        }
      ),
      /* @__PURE__ */ e(
        ga,
        {
          bottomLinks: r,
          copyright: c,
          identifierText: d,
          identifier: u,
          defaultLinkTarget: g,
          hideIdentifier: y
        }
      )
    ] })
  ] })
);
Na.displayName = "Footer";
const zt = D(
  ({
    isOpen: t,
    defaultOpen: a = !1,
    onOpenChange: n,
    buttonText: s,
    buttonClassName: l = "krds-btn small text drop-btn",
    children: o,
    className: r,
    ...c
  }, d) => {
    const [u, g] = ne({
      value: t,
      defaultValue: a,
      finalValue: !1,
      onChange: n
    }), v = j(null), y = j(null), x = (k) => {
      k.preventDefault(), k.stopPropagation(), g(!u);
    };
    return J(() => {
      const k = (E) => {
        y.current && !y.current.contains(E.target) && g(!1);
      };
      return u && document.addEventListener("mousedown", k), () => {
        document.removeEventListener("mousedown", k);
      };
    }, [u, g]), /* @__PURE__ */ h("div", { ref: y, className: m("krds-drop-wrap", u && "active", r), children: [
      /* @__PURE__ */ e(
        "button",
        {
          type: "button",
          className: l,
          onClick: x,
          "aria-expanded": u,
          ref: d,
          ...c,
          children: s
        }
      ),
      /* @__PURE__ */ e("div", { className: "drop-menu", style: { display: u ? "block" : "none" }, ref: v, children: /* @__PURE__ */ e("div", { className: "drop-in", children: o }) })
    ] });
  }
);
zt.displayName = "Dropdown";
const Yt = D(
  ({ children: t, className: a, ...n }, s) => /* @__PURE__ */ e("div", { ref: s, className: m("drop-top", a), ...n, children: t })
);
Yt.displayName = "DropdownTopSection";
const Kt = D(
  ({ children: t, className: a, ...n }, s) => /* @__PURE__ */ e("ul", { ref: s, className: m("drop-list", a), ...n, children: t })
);
Kt.displayName = "DropdownList";
const qt = D(
  ({ children: t, className: a, isActive: n, ...s }, l) => /* @__PURE__ */ e("li", { children: /* @__PURE__ */ h("button", { ref: l, type: "button", className: m("item-link", a), ...s, children: [
    t,
    n && /* @__PURE__ */ e("span", { className: "sr-only", children: "선택됨" })
  ] }) })
);
qt.displayName = "DropdownItem";
const Wt = D(
  ({ children: t, className: a, ...n }, s) => /* @__PURE__ */ e("div", { ref: s, className: m("drop-bottom", a), ...n, children: t })
);
Wt.displayName = "DropdownBottomSection";
const $e = Object.assign(zt, {
  Item: qt,
  List: Kt,
  TopSection: Yt,
  BottomSection: Wt
}), ot = ({
  children: t,
  enabled: a = !0,
  container: n,
  containerClassName: s
}) => {
  const [l, o] = ee(null);
  return It(() => {
    if (!a) return;
    let r = null;
    return n ? typeof n == "string" ? (r = document.getElementById(n), r || (r = document.createElement("div"), r.id = n, s && (r.className = s), document.body.appendChild(r))) : n instanceof HTMLElement && (r = n) : r = document.body, o(r), () => {
      if (r && typeof n == "string" && r.children.length === 0 && r.id === n)
        try {
          document.body.removeChild(r);
        } catch {
        }
    };
  }, [a, n, s]), a ? l ? Jn(t, l) : null : /* @__PURE__ */ e(Ee, { children: t });
};
ot.displayName = "Portal";
const ya = "찾고자 하는 메뉴명을 입력해 주세요", ka = "검색", Mt = "전체메뉴 닫기", ye = (t, a) => t && t.trim().length > 0 ? t : a, ht = (t) => t.replace(/[^a-zA-Z0-9_-]/g, "_"), wt = D(
  ({ label: t, children: a, className: n, onClick: s, ...l }, o) => /* @__PURE__ */ e(
    "button",
    {
      type: "button",
      className: m("btn-navi all", n),
      onClick: s,
      ref: o,
      "aria-controls": "mobile-nav",
      ...l,
      children: a ?? t
    }
  )
);
wt.displayName = "MainMenuMobileTrigger";
const wa = (t) => {
  const a = t.querySelectorAll(
    'a, button, [tabindex="0"], input, textarea, select'
  );
  if (!a.length) return () => {
  };
  const n = a[0], s = a[a.length - 1], l = (o) => {
    if (o.key !== "Tab") return;
    const r = t.ownerDocument.activeElement;
    o.shiftKey ? r === n && (o.preventDefault(), s.focus()) : r === s && (o.preventDefault(), n.focus());
  };
  return t.addEventListener("keydown", l), () => t.removeEventListener("keydown", l);
}, Ca = () => {
  let t = document.querySelector(".gnb-backdrop"), a = !1;
  return t || (t = document.createElement("div"), t.className = "gnb-backdrop", document.body.appendChild(t), a = !0), { backdrop: t, owns: a };
}, Sa = 1024, Ut = D((t, a) => {
  const {
    desktop: n,
    mobile: s,
    mobileTrigger: l,
    onMobileMenuOpen: o,
    onMobileMenuClose: r,
    desktopPortalId: c,
    mobilePortalId: d
  } = t, u = s.id?.trim() ? s.id : "mobile-nav", g = "mobileTriggerPortalId" in t ? t.mobileTriggerPortalId : t.mobileTriggerContainerElement, v = j(null), y = j(null), x = j(null), k = j({}), E = j({}), N = j({}), L = j({}), O = j({}), R = j({}), M = le(
    () => n.items.map((i, p) => ({
      item: i,
      key: ye(
        i.id,
        `desktop-main-${p}`
      ),
      index: p
    })),
    [n.items]
  ), S = le(
    () => s.body.mainItems.map((i, p) => ({
      item: i,
      key: ye(i.id, `mobile-main-${p}`),
      index: p
    })),
    [s.body.mainItems]
  ), [b, T] = ee(null), [B, Y] = ee({}), [V, U] = ee(!1), [G, K] = ee(
    S[0]?.key ?? null
  ), [H, _] = ee({}), [te, ie] = ee(null), [ce, oe] = ee(""), [$, Q] = ee("idle"), me = j(V), de = j(null), ge = j(V), Ne = j(null);
  Gn(a, () => v.current, []), J(() => {
    S.length ? K((i) => i ?? S[0].key) : K(null);
  }, [S]);
  const ke = le(() => {
    const i = /* @__PURE__ */ new Map();
    return M.forEach((p) => i.set(p.key, p.item)), i;
  }, [M]), be = b ? ke.get(b) : null, X = !!(be && be.type === "dropdown" && !be.isDropdownTrigger), ue = F(() => {
    T(null);
  }, []);
  J(() => {
    const { backdrop: i, owns: p } = Ca();
    Ne.current = { backdrop: i, owns: p };
    const f = () => ue();
    return i.addEventListener("click", f), () => {
      i.removeEventListener("click", f), p && i.parentElement === document.body && document.body.removeChild(i);
    };
  }, [ue]), J(() => {
    const i = (p) => {
      const f = v.current;
      if (!f) return;
      const w = p.target;
      f.contains(w) || ue();
    };
    return document.addEventListener("mousedown", i), () => document.removeEventListener("mousedown", i);
  }, [ue]), J(() => {
    const i = (p) => {
      p.code === "Escape" && (ue(), U(!1));
    };
    return document.addEventListener("keyup", i), () => document.removeEventListener("keyup", i);
  }, [ue]), J(() => {
    ge.current = V;
  }, [V]), J(() => {
    const i = Ne.current;
    if (!i) return;
    const { backdrop: p } = i, f = document.body;
    X ? (p.classList.add("active"), f.classList.add("is-gnb-web"), f.scrollHeight > window.innerHeight && f.classList.add("hasScrollY")) : (p.classList.remove("active"), f.classList.remove("is-gnb-web"), f.classList.remove("hasScrollY"));
  }, [X]);
  const Le = F((i, p) => {
    p ? k.current[i] = p : delete k.current[i];
  }, []), Se = F(
    (i, p, f) => {
      const w = `${i}__${p}`;
      f ? E.current[w] = f : delete E.current[w];
    },
    []
  ), we = F((i, p) => {
    p ? N.current[i] = p : delete N.current[i];
  }, []), Fe = F((i, p) => {
    p ? L.current[i] = p : delete L.current[i];
  }, []), _e = F((i, p) => {
    p ? O.current[i] = p : delete O.current[i];
  }, []), ae = F((i, p) => {
    p ? R.current[i] = p : delete R.current[i];
  }, []);
  It(() => {
    if (!b) return;
    const i = k.current[b];
    if (!i) return;
    const p = B[b];
    if (!p) {
      i.style.minHeight = "";
      return;
    }
    const f = `${b}__${p}`, w = E.current[f];
    w && (i.style.minHeight = w.scrollHeight ? `${w.scrollHeight}px` : "");
  }, [b, B]), J(() => {
    b || Object.values(k.current).forEach((i) => {
      i && (i.style.minHeight = "");
    });
  }, [b]);
  const Ce = F(
    (i) => {
      const { item: p, key: f } = i;
      if (p.type === "link") return;
      if (b === f) {
        ue();
        return;
      }
      const w = p, A = w.sections.findIndex((z) => z.type === "menu"), W = A >= 0 ? ye(
        w.sections[A].id,
        `${f}-section-${A}`
      ) : null;
      T(f), Y((z) => ({ ...z, [f]: W }));
    },
    [ue, b]
  ), We = F((i, p) => {
    Y((f) => ({ ...f, [i]: p }));
  }, []), xe = le(() => M.map((i) => i.key), [M]), Ie = F(
    (i, p) => {
      const f = (w) => {
        const A = xe[(w + xe.length) % xe.length];
        N.current[A]?.focus();
      };
      switch (i.key) {
        case "Home":
          i.preventDefault(), f(0);
          break;
        case "End":
          i.preventDefault(), f(xe.length - 1);
          break;
        case "ArrowRight":
        case "ArrowDown":
          i.preventDefault(), f(p + 1);
          break;
        case "ArrowLeft":
        case "ArrowUp":
          i.preventDefault(), f(p - 1);
          break;
      }
    },
    [xe]
  );
  J(() => {
    const i = y.current;
    if (i) {
      if (de.current && (clearTimeout(de.current), de.current = null), V) {
        Q((w) => w === "open" || w === "opening" ? w : "pre-opening");
        const p = i.querySelector(".gnb-main-trigger.active");
        if (p && x.current) {
          const w = p.getAttribute("aria-controls");
          if (w) {
            const A = document.getElementById(w);
            if (A && x.current) {
              const W = x.current;
              W.style.scrollBehavior = "auto", W.scrollTop = A.offsetTop, requestAnimationFrame(() => {
                W.style.scrollBehavior = "";
              });
            }
          }
        }
        const f = setTimeout(() => {
          if (!ge.current) return;
          Q((A) => A === "open" ? A : "opening"), document.body.classList.add("is-gnb-mobile");
          const w = (A) => {
            if (A.target !== i) return;
            if (!ge.current) {
              i.removeEventListener("transitionend", w);
              return;
            }
            Q("open");
            const W = i.querySelector(".gnb-wrap");
            W instanceof HTMLElement && (W.setAttribute("tabindex", "0"), W.focus()), i.removeEventListener("transitionend", w), document.querySelector("#krds-header .header-in")?.setAttribute("inert", ""), document.getElementById("container")?.setAttribute("inert", ""), document.getElementById("footer")?.setAttribute("inert", "");
          };
          i.addEventListener("transitionend", w);
        }, 100);
        de.current = f;
      } else {
        Q((w) => w === "idle" ? w : "closing"), document.querySelector("#krds-header .header-in")?.removeAttribute("inert"), document.getElementById("container")?.removeAttribute("inert"), document.getElementById("footer")?.removeAttribute("inert");
        const p = (w) => {
          if (w.target !== i) return;
          if (ge.current) {
            i.removeEventListener("transitionend", p);
            return;
          }
          Q("idle");
          const A = i.getAttribute("id"), W = A ? document.querySelector(`[aria-controls=${A}]`) : null;
          W instanceof HTMLElement && W.focus(), i.removeAttribute("tabindex"), i.removeEventListener("transitionend", p);
        };
        i.addEventListener("transitionend", p);
        const f = setTimeout(() => {
          document.body.classList.remove("is-gnb-mobile");
        }, 400);
        de.current = f;
      }
      return () => {
        de.current && (clearTimeout(de.current), de.current = null);
      };
    }
  }, [V]), J(() => {
    me.current !== V && (V ? o?.() : r?.(), me.current = V);
  }, [V, r, o]), J(() => () => {
    me.current && r?.();
  }, [r]), J(() => {
    if (!V) return;
    const i = y.current;
    return i ? wa(i) : void 0;
  }, [V]);
  const Ae = F(() => {
    Q((i) => i === "open" || i === "opening" ? i : "pre-opening"), U(!0);
  }, []), je = F(() => {
    Q((i) => i === "idle" ? i : "closing"), U(!1), ie(null);
  }, []), Ge = F((i) => {
    const p = i.target, f = y.current?.querySelector(".gnb-wrap");
    f && !f.contains(p) && f instanceof HTMLElement && f.focus();
  }, []), He = le(() => l && Ke(l) ? yt(l, {
    onClick: (i) => {
      typeof l.props.onClick == "function" && l.props.onClick(i), i.defaultPrevented || Ae();
    },
    "aria-controls": l.props["aria-controls"] ?? u
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
  }) : /* @__PURE__ */ e(wt, { onClick: Ae, "aria-controls": u }), [l, Ae, u]), ze = le(
    () => /* @__PURE__ */ e(ot, { container: g, children: He }),
    [g, He]
  ), Xe = $ !== "idle", Je = $ === "opening" || $ === "open", at = F(
    (i, p) => {
      i.preventDefault();
      const { item: f, key: w } = p;
      K(w);
      const A = f.panels[0];
      if (!A) return;
      const W = ye(A.id, `${w}-panel-0`), z = L.current[W], fe = x.current;
      z && fe && fe.scrollTo({ top: z.offsetTop, left: 0, behavior: "smooth" });
    },
    []
  ), Qe = le(
    () => S.flatMap(
      (i) => i.item.panels.map((p, f) => ({
        panel: p,
        panelKey: ye(p.id, `${i.key}-panel-${f}`),
        ownerKey: i.key
      }))
    ),
    [S]
  );
  J(() => {
    const i = x.current;
    if (!i) return;
    const p = () => {
      let f = null;
      const w = i.scrollTop;
      for (const { panelKey: A, ownerKey: W } of Qe) {
        const z = L.current[A];
        z && w >= z.offsetTop - 10 && (f = W);
      }
      f && K(f);
    };
    return i.addEventListener("scroll", p), () => i.removeEventListener("scroll", p);
  }, [Qe]), J(() => {
    const i = () => {
      window.innerWidth >= Sa && U(!1);
    };
    return window.addEventListener("resize", i), () => window.removeEventListener("resize", i);
  }, []);
  const st = F((i) => {
    _((p) => ({ ...p, [i]: !p[i] }));
  }, []), Pe = F(
    (i, p, f) => {
      i.preventDefault(), ie(f);
      const w = O.current[f];
      w && (w.style.display = "block", requestAnimationFrame(() => {
        w.classList.add("is-open"), w.querySelector(".trigger-prev")?.focus();
      })), R.current[f] = i.currentTarget;
    },
    []
  ), Ye = F((i) => {
    const p = O.current[i], f = R.current[i];
    p && (p.classList.remove("is-open"), setTimeout(() => {
      p.style.display = "", f?.focus();
    }, 250)), ie((w) => w === i ? null : w);
  }, []), Ze = F(
    (i) => {
      i.preventDefault();
      const p = ce.trim();
      p && s.header?.search?.onSearch?.(p);
    },
    [s.header?.search, ce]
  ), rt = (i, p) => {
    const f = i.some((w) => w.description);
    return /* @__PURE__ */ e("ul", { className: m(f && "type-description"), children: i.map((w, A) => {
      const W = ye(w.id, `${p}-leaf-${A}`), z = w.href && !w.asButton ? /* @__PURE__ */ h(
        "a",
        {
          href: w.href || "#",
          target: w.isExternal ? "_blank" : void 0,
          title: w.isExternal ? "새 창 열림" : void 0,
          onClick: w.onClick,
          children: [
            w.label,
            w.isExternal && /* @__PURE__ */ e("i", { className: "svg-icon ico-go" })
          ]
        }
      ) : /* @__PURE__ */ e("button", { type: "button", onClick: w.onClick, children: w.label });
      return w.description ? /* @__PURE__ */ h("li", { children: [
        /* @__PURE__ */ e("h3", { className: "tit", children: z }),
        /* @__PURE__ */ e("p", { className: "txt", children: w.description })
      ] }, W) : /* @__PURE__ */ e("li", { children: z }, W);
    }) });
  }, lt = (i) => i ? /* @__PURE__ */ h("div", { className: "gnb-sub-banner", children: [
    i.badgeText && /* @__PURE__ */ e("span", { className: "krds-badge bg-secondary", children: i.badgeText }),
    i.children,
    /* @__PURE__ */ h("button", { type: "button", className: "krds-btn medium text", onClick: i.onButtonClick, children: [
      i.buttonText,
      " ",
      /* @__PURE__ */ e("i", { className: "svg-icon ico-angle right" })
    ] })
  ] }) : null, I = (i, p) => {
    const { section: f, key: w } = p, A = B[i] === w, W = ht(`gnb-sub-${w}`);
    return /* @__PURE__ */ h("li", { children: [
      /* @__PURE__ */ e(
        "button",
        {
          type: "button",
          className: m("gnb-sub-trigger", A && "active"),
          "data-trigger": "gnb",
          "aria-expanded": A ? "true" : "false",
          "aria-controls": W,
          onClick: () => We(i, w),
          children: f.label
        }
      ),
      /* @__PURE__ */ h(
        "div",
        {
          className: m(
            "gnb-sub-list",
            f.layout && f.layout !== "default" && f.layout,
            A && "active"
          ),
          id: W,
          ref: (z) => Se(i, w, z),
          children: [
            /* @__PURE__ */ h("div", { className: "gnb-sub-content", children: [
              /* @__PURE__ */ h("h2", { className: "sub-title", children: [
                /* @__PURE__ */ e("span", { children: f.label }),
                f.quickLinkHref && /* @__PURE__ */ h(
                  "a",
                  {
                    href: f.quickLinkHref || "#",
                    className: "krds-btn link basic small",
                    target: f.quickLinkExternal ? "_blank" : void 0,
                    title: f.quickLinkExternal ? "새 창 열림" : void 0,
                    children: [
                      /* @__PURE__ */ e("span", { className: "underline", children: f.quickLinkLabel ?? "바로가기" }),
                      /* @__PURE__ */ e("i", { className: "svg-icon ico-angle right" })
                    ]
                  }
                )
              ] }),
              rt(f.items, w)
            ] }),
            lt(f.banner)
          ]
        }
      )
    ] }, w);
  }, q = (i) => {
    const { section: p, key: f } = i;
    return /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e(
      "a",
      {
        href: p.href || "#",
        className: "gnb-sub-trigger is-link",
        "data-trigger": "gnb",
        target: p.isExternal ? "_blank" : void 0,
        title: p.isExternal ? "새 창 열림" : void 0,
        onClick: p.onClick,
        children: p.label
      }
    ) }, f);
  }, he = (i) => {
    const { item: p, key: f, index: w } = i;
    if (p.type === "link") {
      const Z = p, Ue = {
        "data-trigger": "gnb",
        onKeyDown: (et) => Ie(et, w),
        ref: (et) => we(f, et)
      };
      return Z.renderAsButton || !Z.href ? /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e(
        "button",
        {
          type: "button",
          className: "gnb-main-trigger is-link",
          onClick: Z.onClick,
          ...Ue,
          children: Z.label
        }
      ) }, f) : /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e(
        "a",
        {
          href: Z.href || "#",
          className: "gnb-main-trigger is-link",
          target: Z.isExternal ? "_blank" : void 0,
          title: Z.isExternal ? "새 창 열림" : void 0,
          onClick: Z.onClick,
          ...Ue,
          children: Z.label
        }
      ) }, f);
    }
    const A = p, W = b === f, z = ht(`gnb-main-${f}`), fe = A.sections.map((Z, Ue) => ({
      section: Z,
      key: ye(
        Z.id,
        `${f}-section-${Ue}`
      ),
      index: Ue
    }));
    return /* @__PURE__ */ h("li", { children: [
      /* @__PURE__ */ e(
        "button",
        {
          type: "button",
          className: m(
            "gnb-main-trigger",
            A.isDropdownTrigger && "is-dropdown",
            W && "active"
          ),
          "data-trigger": "gnb",
          "aria-haspopup": "true",
          "aria-expanded": W ? "true" : "false",
          "aria-controls": z,
          onClick: () => Ce({ item: p, key: f }),
          onKeyDown: (Z) => Ie(Z, w),
          ref: (Z) => we(f, Z),
          children: A.label
        }
      ),
      /* @__PURE__ */ e("div", { className: m("gnb-toggle-wrap", W && "is-open"), id: z, children: /* @__PURE__ */ e(
        "div",
        {
          className: m("gnb-main-list", A.mainListClassName),
          "data-has-submenu": fe.some((Z) => Z.section.type === "menu") ? "true" : "false",
          ref: (Z) => Le(f, Z),
          children: /* @__PURE__ */ e("ul", { children: fe.map(
            (Z) => Z.section.type === "menu" ? I(f, {
              section: Z.section,
              key: Z.key
            }) : q({
              section: Z.section,
              key: Z.key
            })
          ) })
        }
      ) })
    ] }, f);
  }, pe = (i) => {
    const { item: p, key: f } = i;
    if (p.type === "link")
      return /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e("a", { href: p.href || "#", className: "depth3-trigger", children: p.label }) }, f);
    const w = p, A = te === f, W = w.items.map((z, fe) => ({
      link: z,
      key: ye(z.id, `${f}-depth4-${fe}`)
    }));
    return /* @__PURE__ */ h("li", { children: [
      /* @__PURE__ */ e(
        "a",
        {
          href: "#",
          className: m("depth3-trigger", "has-depth4", A && "active"),
          onClick: (z) => Pe(z, w, f),
          ref: (z) => ae(f, z),
          children: w.label
        }
      ),
      /* @__PURE__ */ h(
        "div",
        {
          className: m("depth4-wrap", A && "is-open"),
          style: { display: A ? "block" : void 0 },
          ref: (z) => _e(f, z),
          children: [
            /* @__PURE__ */ h("div", { className: "depth4-head", children: [
              /* @__PURE__ */ h(
                "button",
                {
                  type: "button",
                  className: "krds-btn icon trigger-prev",
                  onClick: () => Ye(f),
                  children: [
                    /* @__PURE__ */ e("span", { className: "sr-only", children: "이전화면" }),
                    /* @__PURE__ */ e("i", { className: "svg-icon ico-angle left" })
                  ]
                }
              ),
              /* @__PURE__ */ h(
                "button",
                {
                  type: "button",
                  className: "krds-btn icon trigger-close",
                  onClick: () => Ye(f),
                  children: [
                    /* @__PURE__ */ e("span", { className: "sr-only", children: Mt }),
                    /* @__PURE__ */ e("i", { className: "svg-icon ico-popup-close" })
                  ]
                }
              )
            ] }),
            /* @__PURE__ */ h("div", { className: "depth4-body", children: [
              /* @__PURE__ */ e("h4", { className: "sub-title", children: w.title }),
              /* @__PURE__ */ e("ul", { className: "depth4-ul", children: W.map((z) => /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e(
                "a",
                {
                  href: z.link.href || "#",
                  target: z.link.isExternal ? "_blank" : void 0,
                  title: z.link.isExternal ? "새 창 열림" : void 0,
                  children: z.link.label
                }
              ) }, z.key)) })
            ] })
          ]
        }
      )
    ] }, f);
  }, De = (i) => {
    const { item: p, key: f } = i;
    if (p.type === "link") {
      const z = p;
      return /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e("a", { href: z.href || "#", className: "gnb-sub-trigger", children: z.label }) }, f);
    }
    const w = p, A = f, W = H[A] ?? !1;
    return /* @__PURE__ */ h("li", { children: [
      /* @__PURE__ */ e(
        "a",
        {
          href: "#",
          className: m("gnb-sub-trigger", "has-depth3", W && "active"),
          onClick: (z) => {
            z.preventDefault(), st(A);
          },
          children: w.label
        }
      ),
      /* @__PURE__ */ e("div", { className: m("depth3-wrap", W && "is-open"), children: /* @__PURE__ */ e("ul", { children: w.items.map(
        (z, fe) => pe({
          item: z,
          key: ye(z.id, `${A}-child-${fe}`)
        })
      ) }) })
    ] }, A);
  }, se = (i) => {
    const { panel: p, key: f } = i, w = ht(f);
    return /* @__PURE__ */ h(
      "div",
      {
        className: "gnb-sub-list",
        id: w,
        ref: (A) => Fe(f, A),
        children: [
          /* @__PURE__ */ e("h2", { className: "sub-title", children: p.label }),
          /* @__PURE__ */ e("ul", { children: p.items.map(
            (A, W) => De({
              item: A,
              key: ye(
                A.id,
                `${f}-item-${W}`
              )
            })
          ) })
        ]
      },
      f
    );
  }, Be = (i) => {
    const { item: p, key: f } = i, w = p.panels[0], A = w ? ye(w.id, `${f}-panel-0`) : f, W = `#${ht(A)}`;
    return /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e(
      "a",
      {
        href: W,
        className: m("gnb-main-trigger", G === f && "active"),
        onClick: (z) => at(z, i),
        children: p.label
      }
    ) }, f);
  }, Te = (i, p) => {
    const f = ye(i.id, `utility-${p}`);
    return "href" in i && i.href ? /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e("a", { href: i.href || "#", className: "krds-btn xsmall text", onClick: i.onClick, children: i.label }) }, f) : /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e("button", { type: "button", className: "krds-btn xsmall text", onClick: i.onClick, children: i.label }) }, f);
  }, C = /* @__PURE__ */ e("nav", { className: "krds-main-menu", ref: v, children: /* @__PURE__ */ e("div", { className: "inner", children: /* @__PURE__ */ e("ul", { className: "gnb-menu", children: M.map(he) }) }) }), P = /* @__PURE__ */ e(
    "div",
    {
      id: u,
      className: m(
        "krds-main-menu-mobile",
        Je && "is-backdrop",
        Je && "is-open"
      ),
      style: { display: Xe ? "block" : "none" },
      ref: y,
      onClick: Ge,
      children: /* @__PURE__ */ h("div", { className: "gnb-wrap", children: [
        /* @__PURE__ */ h("div", { className: "gnb-header", children: [
          s.header?.utilities && s.header.utilities.length > 0 && /* @__PURE__ */ e("div", { className: "gnb-utils", children: /* @__PURE__ */ e("ul", { className: "utility-list", children: s.header.utilities.map((i, p) => Te(i, p)) }) }),
          s.header?.login && /* @__PURE__ */ e("div", { className: "gnb-login", children: s.header.login.isLoggedIn ? /* @__PURE__ */ h(Ee, { children: [
            s.header.login.userName && /* @__PURE__ */ e("span", { className: "user", children: s.header.login.userName }),
            /* @__PURE__ */ h(
              "button",
              {
                type: "button",
                className: "krds-btn large text",
                onClick: s.header.login.onLogout,
                children: [
                  /* @__PURE__ */ e("i", { className: "svg-icon ico-logout" }),
                  s.header.login.logoutText ?? "로그아웃"
                ]
              }
            )
          ] }) : /* @__PURE__ */ h(
            "button",
            {
              type: "button",
              className: "krds-btn large text",
              onClick: s.header.login.onLogin,
              children: [
                /* @__PURE__ */ e("i", { className: "svg-icon ico-log" }),
                s.header.login.loginText ?? "로그인을 해주세요"
              ]
            }
          ) }),
          s.header?.services && s.header.services.length > 0 && /* @__PURE__ */ e("div", { className: "gnb-service-menu", children: s.header.services.map((i, p) => {
            const f = ye(i.id, `service-${p}`);
            return /* @__PURE__ */ e(
              "a",
              {
                href: i.href || "#",
                className: "link",
                onClick: i.onClick,
                children: i.label
              },
              f
            );
          }) }),
          s.header?.search && /* @__PURE__ */ h("form", { className: "sch-input", onSubmit: Ze, children: [
            /* @__PURE__ */ e(
              "input",
              {
                type: "text",
                className: "krds-input",
                value: ce,
                onChange: (i) => oe(i.target.value),
                placeholder: s.header.search.placeholder ?? ya,
                title: "찾고자 하는 메뉴명 입력"
              }
            ),
            /* @__PURE__ */ h("button", { type: "submit", className: "krds-btn medium icon ico-search", children: [
              /* @__PURE__ */ e("span", { className: "sr-only", children: ka }),
              /* @__PURE__ */ e("i", { className: "svg-icon ico-sch" })
            ] })
          ] })
        ] }),
        /* @__PURE__ */ h("div", { className: "gnb-body", ref: x, children: [
          /* @__PURE__ */ h("div", { className: "gnb-menu", children: [
            /* @__PURE__ */ e("div", { className: "menu-wrap", children: /* @__PURE__ */ e("ul", { children: S.map(Be) }) }),
            /* @__PURE__ */ e("div", { className: "submenu-wrap", children: S.flatMap(
              (i) => i.item.panels.map(
                (p, f) => se({
                  panel: p,
                  key: ye(p.id, `${i.key}-panel-${f}`)
                })
              )
            ) })
          ] }),
          s.bottomLinks && s.bottomLinks.length > 0 && /* @__PURE__ */ e("div", { className: "gnb-bottom", children: s.bottomLinks.map((i, p) => /* @__PURE__ */ h(
            "a",
            {
              href: i.href || "#",
              className: "krds-btn small text",
              target: i.isExternal ? "_blank" : void 0,
              title: i.isExternal ? "새 창 열림" : void 0,
              onClick: i.onClick,
              children: [
                i.label,
                /* @__PURE__ */ e("i", { className: `svg-icon ${i.isExternal ? "ico-go" : "ico-angle right"}` })
              ]
            },
            ye(i.id, `bottom-${p}`)
          )) })
        ] }),
        /* @__PURE__ */ h(
          "button",
          {
            type: "button",
            className: "krds-btn medium icon",
            id: "close-nav",
            onClick: je,
            children: [
              /* @__PURE__ */ e("span", { className: "sr-only", children: Mt }),
              /* @__PURE__ */ e("i", { className: "svg-icon ico-popup-close" })
            ]
          }
        )
      ] })
    }
  );
  return /* @__PURE__ */ h(Ee, { children: [
    ze,
    c ? /* @__PURE__ */ e(ot, { container: c, children: C }) : C,
    d ? /* @__PURE__ */ e(ot, { container: d, children: P }) : P
  ] });
});
Ut.displayName = "MainMenu";
const gt = Ut;
gt.MobileTrigger = wt;
const xa = D(
  ({ children: t, ...a }, n) => Ke(t) ? yt(t, {
    ...a,
    ...t?.props,
    ref: n
  }) : null
);
xa.displayName = "HeaderSlot";
const jt = Ve(null), mt = () => {
  const t = Oe(jt);
  if (!t)
    throw new Error("Header compound components must be used within Header");
  return t;
}, Gt = Ve(null), Da = () => Oe(Gt), Ta = "krds-header", Ea = "krds-header-mobile-trigger", Ma = "krds-header-in", La = "krds-header", Xt = D((t, a) => {
  const {
    children: n,
    id: s,
    className: l,
    mobileMenuTriggerPortalId: o,
    desktopMenuPortalId: r = Ma,
    mobileMenuPortalId: c = La,
    ...d
  } = t, u = s ?? Ta, [g, v] = ee(null), y = o ?? Ea, x = F((N) => {
    v(N);
  }, []), k = F(() => {
    x(null);
  }, [x]);
  J(() => {
    const N = (L) => {
      L.target?.closest(`#${u} .krds-drop-wrap`) || k();
    };
    return document.addEventListener("click", N), () => document.removeEventListener("click", N);
  }, [k, u]), J(() => {
    const N = (L) => {
      L.key === "Escape" && k();
    };
    return document.addEventListener("keyup", N), () => document.removeEventListener("keyup", N);
  }, [k]), J(() => {
    let N = window.scrollY;
    const L = () => {
      const O = document.querySelector("#wrap"), R = document.querySelector("#container");
      if (!O || !R) return;
      const M = R.offsetTop, S = window.scrollY, b = S > N, T = S < N;
      S > M + 50 && b ? (O.classList.add("scroll-down"), O.classList.remove("scroll-up")) : S > M + 50 && T ? (O.classList.add("scroll-up"), O.classList.remove("scroll-down")) : O.classList.remove("scroll-down", "scroll-up"), N = S;
    };
    return window.addEventListener("scroll", L, { passive: !0 }), () => window.removeEventListener("scroll", L);
  }, []);
  const E = le(
    () => ({
      openDropdownId: g,
      setDropdown: x,
      closeDropdowns: k,
      mobileTriggerPortalId: y,
      desktopMenuPortalId: r,
      mobileMenuPortalId: c
    }),
    [
      g,
      x,
      k,
      y,
      r,
      c
    ]
  );
  return /* @__PURE__ */ e(jt.Provider, { value: E, children: /* @__PURE__ */ e("header", { ref: a, id: u, className: l, ...d, children: /* @__PURE__ */ e("div", { id: r, className: "header-in", children: n }) }) });
}), Jt = D(
  ({ children: t, className: a, ...n }, s) => /* @__PURE__ */ e("div", { ref: s, className: m("header-container", a), ...n, children: /* @__PURE__ */ e("div", { className: "inner", children: t }) })
), Qt = ({ children: t, className: a, ...n }) => {
  const s = Xn.map(t, (l) => {
    if (!Ke(l))
      return /* @__PURE__ */ e(nt, { children: l });
    const o = l.type.displayName;
    return o === nt.displayName || o === Ct.displayName ? l : /* @__PURE__ */ e(nt, { children: l });
  });
  return /* @__PURE__ */ e("div", { className: m("header-utility", a), ...n, children: /* @__PURE__ */ e("ul", { className: "utility-list", children: s }) });
}, nt = D(
  ({ className: t, children: a, ...n }, s) => /* @__PURE__ */ e("li", { ref: s, className: t, ...n, children: a })
), Ct = ({
  id: t,
  label: a,
  buttonText: n,
  children: s,
  dropdownClassName: l,
  itemClassName: o,
  buttonClassName: r,
  ...c
}) => {
  const d = Me(), u = t ?? `utility-${d}`, { openDropdownId: g, setDropdown: v, closeDropdowns: y } = mt(), x = g === u, k = F(
    (N) => {
      N ? v(u) : y();
    },
    [y, u, v]
  );
  return /* @__PURE__ */ e(nt, { className: o, children: /* @__PURE__ */ e(
    $e,
    {
      ...c,
      buttonText: n ?? /* @__PURE__ */ h(Ee, { children: [
        a,
        /* @__PURE__ */ e("i", { className: "svg-icon ico-toggle" })
      ] }),
      buttonClassName: m("krds-btn small text drop-btn", r),
      className: m(l),
      isOpen: x,
      onOpenChange: k,
      children: s
    }
  ) });
}, Zt = ({
  logoHref: t = "#",
  onLogoClick: a,
  logoAltText: n = "KRDS - Korea Design System",
  children: s,
  className: l,
  ...o
}) => /* @__PURE__ */ h("div", { className: m("header-branding", l), ...o, children: [
  /* @__PURE__ */ e("h2", { className: "logo", children: t ? /* @__PURE__ */ e("a", { href: t, onClick: a, children: /* @__PURE__ */ e("span", { className: "sr-only", children: n }) }) : /* @__PURE__ */ e("button", { type: "button", onClick: a, children: /* @__PURE__ */ e("span", { className: "sr-only", children: n }) }) }),
  s
] }), en = ({ children: t, className: a, ...n }) => {
  const { mobileTriggerPortalId: s } = mt();
  return /* @__PURE__ */ h("div", { className: m("header-actions", a), ...n, children: [
    t,
    /* @__PURE__ */ e(
      "div",
      {
        id: s,
        className: "header-mobile-trigger",
        "data-mobile-trigger": "true"
      }
    )
  ] });
}, tn = D(
  ({ label: t = "통합검색", title: a = "통합검색 레이어", className: n, type: s = "button", ...l }, o) => /* @__PURE__ */ e(
    "button",
    {
      type: s,
      className: m("btn-navi sch", n),
      title: a,
      ref: o,
      ...l,
      children: t
    }
  )
), nn = D((t, a) => {
  if ("href" in t && t.href) {
    const { href: r, label: c = "로그인", className: d, ...u } = t;
    return /* @__PURE__ */ e(
      "a",
      {
        href: r,
        className: m("btn-navi login", d),
        ref: a,
        ...u,
        children: c
      }
    );
  }
  const { label: n = "로그인", className: s, type: l = "button", ...o } = t;
  return /* @__PURE__ */ e(
    "button",
    {
      type: l,
      className: m("btn-navi login", s),
      ref: a,
      ...o,
      children: n
    }
  );
}), an = D(
  ({ label: t = "회원가입", className: a, type: n = "button", ...s }, l) => /* @__PURE__ */ e("button", { type: n, className: m("btn-navi join", a), ref: l, ...s, children: t })
), sn = D(
  ({ children: t = "로그아웃", className: a, onClick: n, type: s = "button", ...l }, o) => {
    const r = Da(), c = (d) => {
      n?.(d), d.defaultPrevented || r?.close();
    };
    return /* @__PURE__ */ e(
      "button",
      {
        type: s,
        className: m("krds-btn medium text", a),
        ref: o,
        onClick: c,
        ...l,
        children: t
      }
    );
  }
), rn = D(
  ({
    label: t = "나의 GOV",
    name: a,
    remainingTime: n,
    remainingTimeLabel: s = "로그아웃까지 남은 시간",
    extendTimeLabel: l = "시간 연장",
    onExtendTime: o,
    items: r,
    className: c,
    buttonClassName: d,
    dropdownId: u,
    ...g
  }, v) => {
    const y = Me(), x = u ?? `my-gov-${y}`, { openDropdownId: k, setDropdown: E, closeDropdowns: N } = mt(), L = k === x, O = F(
      (b) => {
        b ? E(x) : N();
      },
      [N, x, E]
    ), R = le(
      () => ({
        close: N
      }),
      [N]
    ), { onClick: M, ...S } = g;
    return /* @__PURE__ */ e(Gt.Provider, { value: R, children: /* @__PURE__ */ h(
      $e,
      {
        ref: v,
        isOpen: L,
        onOpenChange: O,
        className: m("my-drop", c),
        buttonText: t,
        buttonClassName: m("btn-navi my drop-btn", d, L && "active"),
        ...S,
        children: [
          /* @__PURE__ */ h($e.TopSection, { children: [
            /* @__PURE__ */ e("p", { className: "my-name", children: a }),
            n && /* @__PURE__ */ h("dl", { className: "my-time", children: [
              /* @__PURE__ */ e("dt", { children: s }),
              /* @__PURE__ */ h("dd", { children: [
                /* @__PURE__ */ e("span", { className: "time", children: n }),
                o && /* @__PURE__ */ e("button", { type: "button", className: "krds-btn medium text", onClick: o, children: l })
              ] })
            ] })
          ] }),
          r && r.length > 0 && /* @__PURE__ */ e($e.List, { children: r.map((b, T) => {
            if (b.href) {
              const {
                key: _,
                label: te,
                className: ie,
                onClick: ce,
                href: oe,
                ...$
              } = b, Q = (me) => {
                ce?.(me), me.defaultPrevented || N();
              };
              return /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e(
                "a",
                {
                  href: oe,
                  className: m("item-link", ie),
                  onClick: Q,
                  ...$,
                  children: te
                }
              ) }, b.key ?? T);
            }
            const {
              key: B,
              label: Y,
              className: V,
              onClick: U,
              type: G = "button",
              ...K
            } = b, H = (_) => {
              U?.(_), _.defaultPrevented || N();
            };
            return /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e(
              "button",
              {
                type: G,
                className: m("item-link", V),
                onClick: H,
                ...K,
                children: Y
              }
            ) }, b.key ?? T);
          }) }),
          /* @__PURE__ */ e($e.BottomSection, { children: /* @__PURE__ */ e(qe.NaviButton.Logout, { onClick: () => console.log("logout") }) })
        ]
      }
    ) });
  }
), ln = D(
  ({ mobileTrigger: t, mobileTriggerProps: a, allMenuButtonLabel: n = "전체메뉴", ...s }, l) => {
    const { mobileTriggerPortalId: o, desktopMenuPortalId: r, mobileMenuPortalId: c } = mt(), d = {
      ...a,
      label: a?.label ?? n
    }, u = /* @__PURE__ */ e(gt.MobileTrigger, { ...d });
    return /* @__PURE__ */ e(
      gt,
      {
        ref: l,
        ...s,
        mobileTrigger: t ?? u,
        mobileTriggerPortalId: o,
        desktopPortalId: r,
        mobilePortalId: c
      }
    );
  }
);
Xt.displayName = "Header";
Jt.displayName = "Header.Container";
Qt.displayName = "Header.Utilities";
nt.displayName = "Header.Utility";
Ct.displayName = "Header.UtilityDropdown";
Zt.displayName = "Header.Branding";
en.displayName = "Header.Navi";
tn.displayName = "Header.NaviButton.Search";
nn.displayName = "Header.NaviButton.Login";
an.displayName = "Header.NaviButton.Join";
sn.displayName = "Header.NaviButton.Logout";
rn.displayName = "Header.NaviButton.MyGov";
ln.displayName = "Header.MainMenu";
const qe = Xt;
qe.Container = Jt;
qe.Utilities = Qt;
qe.Utility = Object.assign(nt, {
  Dropdown: Ct,
  DropdownItem: $e.Item
});
qe.Branding = Zt;
qe.Navi = en;
qe.NaviButton = {
  Search: tn,
  Login: nn,
  Join: an,
  MyGov: rn,
  Logout: sn
};
qe.MainMenu = ln;
const on = D(
  ({ type: t = "decimal", children: a, className: n, ...s }, l) => t === "ordered" ? /* @__PURE__ */ e(
    "ol",
    {
      ref: l,
      className: m("krds-info-list", t, n),
      ...s,
      children: a
    }
  ) : /* @__PURE__ */ e(
    "ul",
    {
      ref: l,
      className: m("krds-info-list", t, n),
      ...s,
      children: a
    }
  )
);
on.displayName = "TextList";
const cn = D(
  ({ children: t, className: a, number: n, ...s }, l) => /* @__PURE__ */ h("li", { ref: l, className: a, ...s, children: [
    n && /* @__PURE__ */ e("span", { className: "num", children: n }),
    t
  ] })
);
cn.displayName = "TextListItem";
const dn = Ve(null), un = () => Oe(dn), Ia = D(
  ({ isOpen: t, defaultOpen: a = !1, onOpenChange: n, children: s, className: l, ...o }, r) => {
    const [c, d] = ne({
      value: t,
      defaultValue: a,
      onChange: n
    }), u = j(null), g = j(null), v = j(!!c), y = Me(), x = Re.Children.toArray(s), k = [], E = x.filter((T) => {
      if (!Re.isValidElement(T))
        return !0;
      const B = T.type, Y = typeof B == "function" || typeof B == "object" ? B.displayName ?? B.name : void 0;
      return B === hn || Y === "HelpPanelTrigger" ? (k.push(T), !1) : !0;
    }), N = F(
      (T) => {
        T && (g.current = document.activeElement), d(T);
      },
      [d]
    ), L = F(() => N(!0), [N]), O = F(() => N(!1), [N]), R = F(() => N(!c), [c, N]);
    J(() => {
      const T = u.current;
      if (!T)
        return;
      const B = !!c;
      if (B && !v.current) {
        g.current || (g.current = document.activeElement);
        const Y = T.querySelectorAll(
          'a, button, [tabindex="0"], input, textarea, select'
        );
        Y.length > 0 && Y[0].focus();
      }
      !B && v.current && (g.current?.focus?.(), g.current = null), v.current = B;
    }, [c]);
    const M = (T) => {
      T.key === "Escape" && O();
    }, S = le(
      () => ({
        isOpen: !!c,
        open: L,
        close: O,
        toggle: R,
        setOpen: N,
        panelId: y,
        lastFocusedElementRef: g
      }),
      [c, L, O, R, N, y]
    ), b = m(
      "krds-help-panel",
      l,
      S.isOpen && "expand"
    );
    return /* @__PURE__ */ h(dn.Provider, { value: S, children: [
      k,
      /* @__PURE__ */ e(
        "div",
        {
          ref: (T) => {
            u.current = T, typeof r == "function" ? r(T) : r && (r.current = T);
          },
          id: y,
          className: b,
          onKeyDown: M,
          ...o,
          children: /* @__PURE__ */ e("div", { className: "help-panel-wrap", children: /* @__PURE__ */ e("div", { className: "help-conts-area", children: E }) })
        }
      )
    ] });
  }
);
Ia.displayName = "HelpPanel";
const hn = D(
  ({ children: t = "도움말", className: a, onClick: n, ...s }, l) => {
    const o = un(), r = m("btn-help-panel", "expand", "btn-help-exec", a);
    return /* @__PURE__ */ h(
      re,
      {
        size: "small",
        variant: "tertiary",
        className: r,
        ref: l,
        onClick: (d) => {
          n?.(d), d.defaultPrevented || o?.open();
        },
        "aria-expanded": o?.isOpen ?? !1,
        "aria-controls": o?.panelId,
        ...s,
        children: [
          /* @__PURE__ */ e("i", { className: "svg-icon ico-fold" }),
          t
        ]
      }
    );
  }
);
hn.displayName = "HelpPanelTrigger";
const Pa = D(
  ({ children: t, srOnlyTitle: a, className: n, ...s }, l) => {
    const o = m("help-conts-area-inner", n);
    return /* @__PURE__ */ h("div", { className: o, ref: l, ...s, children: [
      a && /* @__PURE__ */ e("h3", { className: "sr-only", children: a }),
      t
    ] });
  }
);
Pa.displayName = "HelpPanelContent";
const Aa = D(
  ({ title: t, description: a, children: n, className: s, ...l }, o) => {
    const r = m("conts-area", "help-conts", s);
    return /* @__PURE__ */ e("div", { className: r, ref: o, ...l, children: /* @__PURE__ */ h("div", { className: "conts-wrap", children: [
      /* @__PURE__ */ h("h4", { className: "help-title", children: [
        t,
        /* @__PURE__ */ h("span", { className: "krds-btn medium icon", children: [
          /* @__PURE__ */ e("span", { className: "sr-only", children: "도움말" }),
          /* @__PURE__ */ e("i", { className: "svg-icon ico-help" })
        ] })
      ] }),
      a && /* @__PURE__ */ e("div", { className: "conts-desc", children: /* @__PURE__ */ e("p", { children: a }) }),
      n
    ] }) });
  }
);
Aa.displayName = "HelpSection";
const Ba = D(
  ({ links: t, iconPosition: a = "right", className: n, ...s }, l) => {
    const o = m("link-list", n);
    return /* @__PURE__ */ e("ul", { className: o, ref: l, ...s, children: t.map((r, c) => /* @__PURE__ */ e("li", { children: /* @__PURE__ */ h(
      re,
      {
        as: "a",
        size: "xsmall",
        variant: "link",
        href: r.href,
        target: r.target,
        title: r.target === "_blank" ? "새 창 열림" : void 0,
        className: "basic",
        children: [
          a === "left" && r.icon && /* @__PURE__ */ e("i", { className: `svg-icon ${r.icon}` }),
          r.text,
          a === "right" && /* @__PURE__ */ e("i", { className: `svg-icon ${r.icon || "ico-go"}` })
        ]
      }
    ) }, c)) });
  }
);
Ba.displayName = "HelpLinkList";
const $a = D(
  ({ children: t = "접어두기", className: a, onClick: n, ...s }, l) => {
    const o = un(), r = m("btn-help-panel", "fold", a);
    return /* @__PURE__ */ h(
      re,
      {
        size: "small",
        variant: "tertiary",
        className: r,
        ref: l,
        onClick: (d) => {
          n?.(d), d.defaultPrevented || o?.close();
        },
        "aria-controls": o?.panelId,
        ...s,
        children: [
          /* @__PURE__ */ e("span", { className: "sr-only", children: "도움말" }),
          t,
          /* @__PURE__ */ e("i", { className: "svg-icon ico-fold right" })
        ]
      }
    );
  }
);
$a.displayName = "HelpPanelClose";
const Ra = D(({ children: t, className: a, ...n }, s) => {
  const l = m("conts-area", "related-service", a);
  return /* @__PURE__ */ e("div", { className: l, ref: s, ...n, children: t });
});
Ra.displayName = "HelpRelatedService";
const Oa = D(({ title: t, children: a, className: n, ...s }, l) => {
  const o = m("conts-wrap", n);
  return /* @__PURE__ */ h("div", { className: o, ref: l, ...s, children: [
    /* @__PURE__ */ e("h4", { className: "help-title", children: t }),
    a
  ] });
});
Oa.displayName = "HelpServiceGroup";
const Ha = D(({ title: t, href: a, className: n, ...s }, l) => {
  const o = m("help-title", n);
  return /* @__PURE__ */ e("h4", { className: o, ref: l, ...s, children: a ? /* @__PURE__ */ e("a", { href: a, title: "이전으로 돌아가기", children: t }) : t });
});
Ha.displayName = "HelpTutorialTitle";
const Va = D(({ children: t, className: a, ...n }, s) => {
  const l = m("coach-help-process", a);
  return /* @__PURE__ */ e("ul", { className: l, ref: s, ...n, children: t });
});
Va.displayName = "HelpCoachProcess";
const Fa = D(({ title: t, isCurrent: a = !1, expandText: n, steps: s, className: l, ...o }, r) => {
  const c = m("tit", a && "current");
  return /* @__PURE__ */ h("li", { className: l, ref: r, ...o, children: [
    /* @__PURE__ */ e("h4", { className: c, children: t }),
    /* @__PURE__ */ e(kt, { buttonText: n, className: "conts-expand-area", children: /* @__PURE__ */ e(on, { type: "decimal", children: s.map((d, u) => /* @__PURE__ */ e(cn, { children: d }, u)) }) })
  ] });
});
Fa.displayName = "HelpCoachTask";
const _a = D(({ children: t, className: a, ...n }, s) => {
  const l = m("help-panel-action", a);
  return /* @__PURE__ */ e("div", { className: l, ref: s, ...n, children: t });
});
_a.displayName = "HelpPanelAction";
const za = D(({ children: t, className: a, ...n }, s) => {
  const l = m("conts-area", a);
  return /* @__PURE__ */ e("div", { className: l, ref: s, ...n, children: t });
});
za.displayName = "HelpContentArea";
const Ya = D(
  ({
    caption: t = "이 페이지의 구성",
    title: a,
    items: n = [],
    actionText: s,
    onActionClick: l,
    infoText: o,
    scrollDown: r = !1,
    className: c,
    scrollContainerRef: d,
    ...u
  }, g) => {
    const [v, y] = ee(""), x = s || o, k = n.length > 0, E = () => {
      const S = document.querySelector("#krds-masthead")?.clientHeight || 0, b = document.querySelector("#krds-header .header-in")?.clientHeight || 0;
      return S + b;
    }, N = (S, b) => {
      b.preventDefault();
      const T = E(), B = d?.current ?? window;
      if (B === window) {
        const U = S.getBoundingClientRect().top + window.scrollY - T;
        window.scrollTo({
          left: 0,
          top: U,
          behavior: "smooth"
        });
      } else {
        const V = B, U = V.getBoundingClientRect(), K = S.getBoundingClientRect().top - U.top, H = V.scrollTop + K - T;
        V.scrollTo({
          left: 0,
          top: H,
          behavior: "smooth"
        });
      }
      if ("key" in b && (b.key === "Enter" || b.key === " ")) {
        const V = S.querySelector(".sec-tit");
        V && (V.setAttribute("tabindex", "-1"), V.focus({ preventScroll: !0 }));
      }
    }, L = (S, b) => {
      const T = document.querySelector(S);
      T && N(T, b);
    }, O = (S, b) => {
      (b.key === "Enter" || b.key === " ") && (b.preventDefault(), L(S, b));
    }, R = () => {
      if (!k) return;
      const S = d?.current ?? window, b = S === window, T = b ? window.scrollY : S.scrollTop, B = b ? window.innerHeight : S.clientHeight, Y = b ? document.body.scrollHeight : S.scrollHeight, V = Math.ceil(B / 5), U = T + B, G = n.map((te) => document.querySelector(te.href)).filter(Boolean);
      if (G.length === 0) return;
      const H = G[0].offsetTop;
      let _ = "";
      if (U >= Y)
        _ = n[n.length - 1]?.href || "";
      else if (T <= H)
        _ = n[0]?.href || "";
      else
        for (let te = G.length - 1; te >= 0; te--) {
          const ie = G[te], ce = ie.offsetHeight, oe = ie.offsetTop - V;
          if (T > oe && T <= oe + ce) {
            _ = n[te]?.href || "";
            break;
          }
        }
      _ !== v && y(_);
    };
    return J(() => {
      if (!k) return;
      R();
      const S = () => {
        R();
      }, b = d?.current ?? window;
      return b.addEventListener("scroll", S, { passive: !0 }), () => {
        b.removeEventListener("scroll", S);
      };
    }, [k, n, v]), /* @__PURE__ */ e(({ children: S }) => r ? /* @__PURE__ */ e("div", { className: "scroll-down", children: S }) : /* @__PURE__ */ e(Ee, { children: S }), { children: /* @__PURE__ */ e("div", { ref: g, className: "krds-in-page-navigation-type", children: /* @__PURE__ */ h("div", { className: m("krds-in-page-navigation-area", c), ...u, children: [
      /* @__PURE__ */ h("div", { className: "in-page-navigation-header", children: [
        /* @__PURE__ */ e("p", { className: "quick-caption", children: t }),
        a && /* @__PURE__ */ e("p", { className: "quick-title", children: a })
      ] }),
      k && /* @__PURE__ */ e("nav", { className: "in-page-navigation-list", children: /* @__PURE__ */ e("ul", { children: n.map((S, b) => /* @__PURE__ */ e("li", { children: /* @__PURE__ */ e(
        "a",
        {
          href: S.href,
          className: m(
            (S.active || v === S.href) && "active"
          ),
          onClick: (T) => L(S.href, T),
          onKeyDown: (T) => O(S.href, T),
          tabIndex: 0,
          children: S.label
        }
      ) }, b)) }) }),
      x && /* @__PURE__ */ h("div", { className: "in-page-navigation-action", children: [
        s && /* @__PURE__ */ e(re, { className: "krds-btn medium", onClick: l, children: s }),
        o && /* @__PURE__ */ e("p", { className: "quick-info", children: o })
      ] })
    ] }) }) });
  }
);
Ya.displayName = "InPageNavigation";
function Ka(t, a) {
  typeof t == "function" ? t(a) : t != null && (t.current = a);
}
function it(...t) {
  return le(() => t.every((a) => a === null) ? null : (a) => {
    for (const n of t)
      Ka(n, a);
  }, [t]);
}
const mn = Ve(null);
function ct() {
  const t = Oe(mn);
  if (!t) throw new Error("LanguageSwitcher.* must be used within <LanguageSwitcher>");
  return t;
}
const Lt = D(
  ({
    value: t,
    defaultValue: a,
    options: n = [],
    onChange: s,
    open: l,
    defaultOpen: o,
    onOpenChange: r,
    closeOnClickOutside: c = !0,
    children: d,
    className: u,
    ...g
  }, v) => {
    const [y, x] = ne({
      value: t,
      defaultValue: a,
      onChange: s,
      finalValue: ""
    }), [k, E] = ne({
      value: l,
      defaultValue: o,
      onChange: r
    }), N = j(null), L = j(null), O = it(v, N), R = j(/* @__PURE__ */ new Map()), M = F(
      (T) => {
        T.key === "Escape" && k && (E(!1), console.log(L.current), L.current?.focus());
      },
      [k, E]
    );
    J(() => {
      if (!c) return;
      const T = (B) => {
        N.current && !N.current.contains(B.target) && k && E(!1);
      };
      return document.addEventListener("mousedown", T), () => document.removeEventListener("mousedown", T);
    }, [k, E, c]);
    const S = m("krds-drop-wrap", "krds-language", u), b = le(
      () => ({
        value: y,
        onChange: x,
        options: n,
        open: k,
        setOpen: E,
        buttonRef: L,
        itemRefs: R
      }),
      [y, x, n, k, E, L]
    );
    return /* @__PURE__ */ e(mn.Provider, { value: b, children: /* @__PURE__ */ e("div", { ref: O, className: S, onKeyDown: M, ...g, children: d }) });
  }
), qa = D(
  ({ label: t = "Language", className: a }, n) => {
    const { open: s, setOpen: l, value: o, options: r, buttonRef: c } = ct(), d = r.find((g) => g.value === o)?.label || o || t, u = it(n, c);
    return /* @__PURE__ */ h(
      re,
      {
        ref: u,
        variant: "text",
        size: "small",
        onClick: () => l(!s),
        "aria-label": t,
        "aria-expanded": s,
        "aria-controls": "language-dropdown",
        "aria-haspopup": "listbox",
        className: m("drop-btn", s && "active", a),
        children: [
          /* @__PURE__ */ e("i", { className: "svg-icon ico-global" }),
          " " + d + " ",
          /* @__PURE__ */ e("i", { className: "svg-icon ico-angle" })
        ]
      }
    );
  }
), Wa = D(
  ({ className: t, children: a }, n) => {
    const { open: s } = ct(), l = j(null), o = it(n, l);
    return J(() => {
      if (s && l.current) {
        const r = l.current, c = r.closest(".krds-drop-wrap");
        if (c) {
          c.classList.remove("drop-left", "drop-right");
          const d = r.getBoundingClientRect(), u = window.innerWidth;
          d.left < 0 ? c.classList.add("drop-left") : u < d.left + d.width && c.classList.add("drop-right");
        }
      }
    }, [s]), /* @__PURE__ */ e(
      "div",
      {
        ref: o,
        className: m("drop-menu", t),
        style: { display: s ? "block" : "none" },
        role: "listbox",
        id: "language-dropdown",
        children: /* @__PURE__ */ e("div", { className: "drop-in", children: a })
      }
    );
  }
), Ua = ({ label: t, className: a }) => {
  const { value: n, options: s } = ct(), l = s.find((o) => o.value === n)?.label || n;
  return /* @__PURE__ */ e("div", { className: "drop-top", children: /* @__PURE__ */ h("p", { className: m("current-laguage", a), children: [
    /* @__PURE__ */ e("span", { children: t }),
    /* @__PURE__ */ e("strong", { children: l })
  ] }) });
}, ja = ({ className: t, ...a }) => {
  const { options: n } = ct();
  return /* @__PURE__ */ e("ul", { className: m("drop-list", t), role: "list", ...a, children: n.map((s) => /* @__PURE__ */ e(pn.OptionItem, { ...s }, s.value)) });
}, Ga = ({
  value: t,
  label: a,
  href: n,
  external: s,
  className: l,
  ...o
}) => {
  const { value: r, onChange: c, setOpen: d, itemRefs: u } = ct(), g = r === t, v = (x) => {
    !n && !s && x.preventDefault(), c(t, { value: t, label: a, href: n, external: s }), d(!1);
  }, y = (x) => {
    x.preventDefault(), u.current?.forEach((k) => {
      k?.style.removeProperty("position"), k?.style.removeProperty("z-index");
    }), u.current?.get(t)?.style.setProperty("position", "relative"), u.current?.get(t)?.style.setProperty("z-index", "1");
  };
  return /* @__PURE__ */ e("li", { role: "listitem", ...o, children: /* @__PURE__ */ h(
    "a",
    {
      ref: (x) => {
        x ? u.current?.set(t, x) : u.current?.delete(t);
      },
      href: n || "#",
      className: m("item-link", g && "active", l),
      onClick: v,
      onFocus: y,
      role: "option",
      "aria-selected": g,
      lang: t,
      ...s ? { target: "_blank", rel: "noopener noreferrer" } : {},
      children: [
        a,
        s && /* @__PURE__ */ e("i", { className: "svg-icon ico-go" })
      ]
    }
  ) });
}, pn = Object.assign(Lt, {
  Root: Lt,
  Trigger: qa,
  Menu: Wa,
  Current: Ua,
  OptionList: ja,
  OptionItem: Ga
});
pn.displayName = "LanguageSwitcher";
const Xa = D(
  ({
    variant: t = "default",
    underline: a = "always",
    preserveColorOnHover: n = !1,
    icon: s,
    external: l = !1,
    size: o = "large",
    disabled: r = !1,
    children: c,
    className: d,
    href: u,
    target: g,
    rel: v,
    onClick: y,
    ...x
  }, k) => {
    const E = m(
      "krds-btn",
      "link",
      o,
      t === "basic" && "basic",
      n && "pure",
      r && "disabled",
      d
    ), N = {
      always: "underline",
      hover: "underline hidden-underline",
      none: ""
    }[a], L = (b) => {
      if (r) {
        b.preventDefault();
        return;
      }
      y?.(b);
    }, O = l ? "_blank" : g, R = l ? "noopener noreferrer" : v, M = l && !x.title ? "새 창 열림" : x.title;
    return /* @__PURE__ */ h(
      "a",
      {
        ref: k,
        className: E,
        href: r ? void 0 : u,
        target: O,
        rel: R,
        title: M,
        onClick: L,
        tabIndex: r ? -1 : void 0,
        "aria-disabled": r,
        ...x,
        children: [
          /* @__PURE__ */ e("span", { className: N, children: c }),
          s || (l ? /* @__PURE__ */ e("i", { className: "svg-icon ico-go" }) : null)
        ]
      }
    );
  }
);
Xa.displayName = "Link";
const Ja = D(
  ({ text: t, className: a, ...n }, s) => /* @__PURE__ */ e("div", { id: "krds-masthead", ref: s, className: a, ...n, children: /* @__PURE__ */ e("div", { className: "toggle-wrap", children: /* @__PURE__ */ e("div", { className: "toggle-head", children: /* @__PURE__ */ e("div", { className: "inner", children: /* @__PURE__ */ e("span", { className: "nuri-txt", children: t }) }) }) }) })
);
Ja.displayName = "Masthead";
const fn = Re.createContext(null), dt = () => {
  const t = Re.useContext(fn);
  if (!t) throw new Error("Modal compound components must be used within <Modal.Root>");
  return t;
}, St = Re.forwardRef(
  ({ children: t, ...a }, n) => Re.isValidElement(t) ? Re.cloneElement(t, {
    ...a,
    ...t?.props,
    ref: n
  }) : null
);
St.displayName = "Slot";
const bt = ({
  children: t,
  size: a = "lg",
  variant: n = "default",
  open: s,
  defaultOpen: l = !1,
  onOpenChange: o,
  closeOnEsc: r = !1,
  closeOnOverlayClick: c = !1,
  usePortal: d = !1,
  portalContainer: u
}) => {
  const [g, v] = ne({
    value: s,
    defaultValue: l,
    onChange: o
  }), y = le(
    () => ({
      open: g,
      setOpen: v,
      size: a,
      variant: n,
      closeOnEsc: r,
      closeOnOverlayClick: c,
      usePortal: d,
      portalContainer: u
    }),
    [g, v, a, n, r, c, d, u]
  );
  return /* @__PURE__ */ e(fn.Provider, { value: y, children: t });
};
bt.displayName = "ModalRoot";
const gn = D(
  ({ onClick: t, asChild: a = !1, children: n, ...s }, l) => {
    const { setOpen: o } = dt(), r = F(
      (c) => {
        t?.(c), o(!0);
      },
      [t, o]
    );
    return a ? /* @__PURE__ */ e(St, { onClick: r, ...s, ref: l, children: n }) : /* @__PURE__ */ e("button", { type: "button", onClick: r, ...s, ref: l, children: n });
  }
);
gn.displayName = "ModalTrigger";
const bn = D(
  ({ className: t = "", ...a }, n) => {
    const { open: s } = dt();
    return /* @__PURE__ */ e("div", { className: m("modal-back", s && "in", t), ...a, ref: n });
  }
);
bn.displayName = "ModalOverlay";
const vn = D(
  ({ className: t = "", children: a, id: n, role: s = "dialog", ...l }, o) => {
    const {
      open: r,
      setOpen: c,
      size: d,
      variant: u,
      closeOnEsc: g,
      closeOnOverlayClick: v,
      usePortal: y,
      portalContainer: x
    } = dt(), k = Me(), E = n || `krds-modal-${k}`, N = j(null), L = j(null), O = j(null), R = j([]), M = j(!1), S = j(null), b = j(null), T = F(() => {
      if (!N.current) return [];
      const K = [
        "a[href]",
        "button:not([disabled])",
        "textarea:not([disabled])",
        "input:not([disabled])",
        "select:not([disabled])",
        '[tabindex]:not([tabindex="-1"])',
        '[contenteditable="true"]',
        "iframe",
        "object",
        "embed",
        "area[href]",
        "audio[controls]",
        "video[controls]"
      ].join(", ");
      return Array.from(N.current.querySelectorAll(K)).filter(
        (H) => {
          const _ = H.getAttribute("aria-hidden");
          return !(H.hasAttribute("disabled") || _ === "true");
        }
      );
    }, []), B = F(() => {
      if (M.current) return;
      const K = T();
      R.current = K, M.current = !0, b.current = setTimeout(() => {
        const H = K[0];
        H && H.focus();
      }, 200);
    }, [T]), Y = F(
      (K) => {
        if (o && (typeof o == "function" ? o(K) : o.current = K), N.current = K, K && r) {
          const H = new MutationObserver(() => {
            B(), H.disconnect();
          });
          return H.observe(K, {
            attributes: !0,
            attributeFilter: ["tabindex"],
            subtree: !0,
            childList: !0
          }), requestAnimationFrame(() => {
            B();
          }), () => H.disconnect();
        }
      },
      [o, r, B]
    ), V = F((K) => {
      L.current = K;
    }, []);
    J(() => () => {
      R.current = [], M.current = !1, S.current && clearTimeout(S.current), b.current && clearTimeout(b.current);
    }, []), J(() => {
      if (!r) {
        R.current = [], M.current = !1, S.current && clearTimeout(S.current), b.current && clearTimeout(b.current);
        return;
      }
      const K = (H) => {
        if ((H.key === "Escape" || H.key === "Esc") && g) {
          c(!1);
          return;
        }
        if (H.key === "Tab") {
          const _ = R.current;
          if (_.length === 0) {
            H.preventDefault();
            return;
          }
          const te = _[0], ie = _[_.length - 1], ce = document.activeElement;
          H.shiftKey && ce === te ? (H.preventDefault(), ie.focus()) : !H.shiftKey && ce === ie && (H.preventDefault(), te.focus());
        }
      };
      return document.addEventListener("keydown", K, !0), () => {
        document.removeEventListener("keydown", K, !0);
      };
    }, [r, g, c]), J(() => {
      if (r) {
        document.body.classList.add("scroll-no"), O.current = document.activeElement;
        const K = document.getElementById("wrap");
        K && K.setAttribute("inert", ""), S.current = setTimeout(() => {
          L.current?.classList.add("in");
        }, 150);
      } else {
        document.body.classList.remove("scroll-no"), O.current?.focus();
        const K = document.getElementById("wrap");
        K && K.removeAttribute("inert");
      }
    }, [r]);
    const U = F(
      (K) => {
        const H = K.target;
        N.current?.contains(H) || (v ? c(!1) : R.current[0]?.focus());
      },
      [c, v]
    ), G = m("krds-modal shown", r && "in", t);
    return /* @__PURE__ */ e(ot, { enabled: y, container: x, children: /* @__PURE__ */ h(
      "section",
      {
        id: E,
        "aria-modal": "true",
        role: s,
        className: G,
        "data-type": u === "default" ? void 0 : u,
        onClick: U,
        ...l,
        children: [
          /* @__PURE__ */ e(
            "div",
            {
              className: m(
                "modal-dialog",
                d === "sm" && "modal-sm",
                d === "md" && "modal-md",
                d === "lg" && "modal-lg"
              ),
              ref: V,
              children: /* @__PURE__ */ h("div", { className: "modal-content", ref: Y, children: [
                a,
                /* @__PURE__ */ e(Qa, {})
              ] })
            }
          ),
          /* @__PURE__ */ e(bn, {})
        ]
      }
    ) });
  }
);
vn.displayName = "ModalContent";
const Nn = D(
  ({ children: t, title: a, titleId: n, className: s = "", ...l }, o) => {
    const r = n;
    return /* @__PURE__ */ e("div", { className: m("modal-header", s), ...l, ref: o, children: a ? /* @__PURE__ */ e("h2", { id: r, className: "modal-title", children: a }) : t });
  }
);
Nn.displayName = "ModalHeader";
const yn = D(
  ({ className: t = "", ...a }, n) => /* @__PURE__ */ e("h2", { className: m("modal-title", t), ...a, ref: n })
);
yn.displayName = "ModalTitle";
const kn = D(
  ({ className: t = "", ...a }, n) => /* @__PURE__ */ e("p", { className: m("text-sm text-muted-foreground", t), ...a, ref: n })
);
kn.displayName = "ModalDescription";
const wn = D(
  ({ children: t, className: a = "", descriptionId: n, ...s }, l) => {
    const [o, r] = ee(!1), c = j(null), d = F(
      (u) => {
        if (l && (typeof l == "function" ? l(u) : l.current = u), c.current = u, u) {
          const g = u.scrollHeight > u.clientHeight;
          r(g);
        }
      },
      [l]
    );
    return /* @__PURE__ */ e(
      "div",
      {
        className: m("modal-conts", a),
        id: n,
        ref: d,
        ...o && { tabIndex: 0 },
        ...s,
        children: /* @__PURE__ */ e("div", { className: "conts-area", children: t })
      }
    );
  }
);
wn.displayName = "ModalBody";
const Cn = D(
  ({ children: t, className: a = "", ...n }, s) => /* @__PURE__ */ e("div", { className: m("modal-btn", "btn-wrap", a), ...n, ref: s, children: t })
);
Cn.displayName = "ModalFooter";
const Qa = () => {
  const { setOpen: t } = dt();
  return /* @__PURE__ */ h(
    re,
    {
      variant: "icon",
      type: "button",
      size: "medium",
      className: m("btn-close"),
      onClick: () => t(!1),
      children: [
        /* @__PURE__ */ e("span", { className: "sr-only", children: "모달 닫기" }),
        /* @__PURE__ */ e("i", { className: "svg-icon ico-popup-close" })
      ]
    }
  );
}, Za = D(
  ({ onClick: t, className: a, children: n, asChild: s = !1, ...l }, o) => {
    const { setOpen: r } = dt(), c = F(
      (u) => {
        t?.(u), r(!1);
      },
      [t, r]
    );
    return /* @__PURE__ */ e(
      s ? St : "button",
      {
        ref: o,
        className: a,
        onClick: c,
        ...l,
        children: n
      }
    );
  }
), es = Object.assign(bt, {
  Root: bt,
  Trigger: gn,
  Content: vn,
  Header: Nn,
  Title: yn,
  Description: kn,
  Body: wn,
  Footer: Cn,
  Close: Za
});
es.displayName = "Modal";
const ts = D(
  ({
    totalPages: t,
    currentPage: a,
    defaultPage: n = 1,
    onChange: s,
    prevLabel: l = "이전",
    nextLabel: o = "다음",
    boundaryCount: r = 1,
    siblingCount: c = 1,
    disabled: d = !1,
    className: u = "",
    ...g
  }, v) => {
    const [y, x] = ne({
      value: a,
      defaultValue: n,
      finalValue: 1,
      onChange: s
    }), k = (b) => Math.max(1, Math.min(t, b)), N = (() => {
      if (t <= 0) return [];
      const b = Array.from(
        { length: Math.min(r, t) },
        (H, _) => _ + 1
      ), T = Array.from(
        { length: Math.min(r, t) },
        (H, _) => t - _
      ).reverse(), B = Math.max(
        Math.min(y - c, t - r - c * 2 - 1),
        r + 2
      ), Y = Math.min(
        Math.max(y + c, r + c * 2 + 2),
        T.length > 0 ? T[0] - 2 : t - 1
      ), V = [];
      V.push(...b), B > r + 2 ? V.push("dots") : r + 1 < t - r && V.push(r + 1);
      for (let H = B; H <= Y; H += 1)
        H >= 1 && H <= t && V.push(H);
      Y < t - r - 1 ? V.push("dots") : t - r > r && V.push(t - r);
      for (const H of T)
        V.includes(H) || V.push(H);
      const U = [], G = /* @__PURE__ */ new Set();
      for (const H of V) {
        const _ = String(H);
        G.has(_) || (G.add(_), U.push(H));
      }
      U.sort((H, _) => H === "dots" || _ === "dots" ? 0 : H - _);
      const K = [];
      for (let H = 0; H < U.length; H++) {
        const _ = U[H];
        if (K.push(_), typeof _ == "number" && H < U.length - 1) {
          const te = U[H + 1];
          typeof te == "number" && te - _ > 1 && K.push("dots");
        }
      }
      return K;
    })(), L = () => {
      if (d) return;
      const b = k(y - 1);
      b !== y && x(b);
    }, O = () => {
      if (d) return;
      const b = k(y + 1);
      b !== y && x(b);
    }, R = (b) => () => {
      d || b !== y && x(b);
    }, M = d || y <= 1, S = d || y >= t;
    return /* @__PURE__ */ h("div", { className: m("krds-pagination", u), ref: v, ...g, children: [
      /* @__PURE__ */ e(
        "span",
        {
          className: m("page-navi prev", M && "disabled"),
          "aria-disabled": M,
          role: "button",
          tabIndex: M ? -1 : 0,
          onClick: M ? void 0 : L,
          children: l
        }
      ),
      /* @__PURE__ */ e("div", { className: "page-links", children: N.map((b, T) => {
        if (b === "dots")
          return /* @__PURE__ */ e("span", { className: "page-link link-dot", "aria-hidden": !0 }, `dots-${T}`);
        const B = b === y;
        return /* @__PURE__ */ e(
          "a",
          {
            className: m("page-link", B && "active"),
            "aria-current": B ? "page" : void 0,
            onClick: R(b),
            href: "#",
            children: B ? /* @__PURE__ */ h(Ee, { children: [
              /* @__PURE__ */ e("span", { className: "sr-only", children: "현재페이지 " }),
              b
            ] }) : b
          },
          b
        );
      }) }),
      /* @__PURE__ */ e(
        "a",
        {
          className: m("page-navi next", S && "disabled"),
          "aria-disabled": S,
          role: "button",
          tabIndex: S ? -1 : 0,
          onClick: S ? void 0 : O,
          href: "#",
          children: o
        }
      )
    ] });
  }
);
ts.displayName = "Pagination";
const xt = Ve(void 0), ns = D(
  ({
    id: t,
    name: a,
    defaultChecked: n,
    checked: s,
    disabled: l = !1,
    size: o = "medium",
    children: r,
    description: c,
    onChange: d,
    className: u,
    value: g,
    ...v
  }, y) => {
    const x = m("krds-form-check", o, u), k = Oe(xt), E = le(() => t ?? `radio-${Math.random().toString(36).substr(2, 9)}`, [t]), N = !!k, L = k?.value, O = k?.name, R = k?.onChange, M = le(() => N ? O : a || `radio-${E}`, [N, O, a, E]), S = N ? L === g : s, [b, T] = ne({
      value: S,
      defaultValue: n,
      onChange: (Y) => {
        d && d({
          target: { checked: Y }
        });
      }
    });
    return /* @__PURE__ */ h("div", { className: x, children: [
      /* @__PURE__ */ e(
        "input",
        {
          type: "radio",
          id: E,
          name: M,
          checked: b,
          defaultChecked: n,
          disabled: l,
          onChange: (Y) => {
            N && g !== void 0 ? typeof g == "string" && R?.(g) : T(Y.target.checked);
          },
          ref: y,
          ...v
        }
      ),
      /* @__PURE__ */ e("label", { htmlFor: E, children: r }),
      c && /* @__PURE__ */ e("div", { className: "krds-form-check-cnt", children: /* @__PURE__ */ e("p", { className: "krds-form-check-p", children: c }) })
    ] });
  }
);
ns.displayName = "Radio";
const as = D(
  ({
    id: t,
    name: a,
    defaultChecked: n,
    checked: s,
    disabled: l = !1,
    size: o = "medium",
    children: r,
    onChange: c,
    className: d,
    value: u,
    ...g
  }, v) => {
    const [y, x] = ee(!1), k = m("krds-form-chip", o, y && "focus", d), E = Oe(xt), N = le(
      () => t ?? `radio-chip-${Math.random().toString(36).substr(2, 9)}`,
      [t]
    ), L = !!E, O = E?.value, R = E?.name, M = E?.onChange, S = le(() => L ? R : a || `radio-chip-${N}`, [L, R, a, N]), b = L ? O === u : s, [T, B] = ne({
      value: b,
      defaultValue: n,
      onChange: (G) => {
        c && c({
          target: { checked: G }
        });
      }
    });
    return /* @__PURE__ */ h("div", { className: k, children: [
      /* @__PURE__ */ e(
        "input",
        {
          type: "radio",
          className: "radio",
          id: N,
          name: S,
          checked: T,
          defaultChecked: n,
          disabled: l,
          onChange: (G) => {
            L && u !== void 0 ? typeof u == "string" && M?.(u) : B(G.target.checked);
          },
          onFocus: () => {
            x(!0);
          },
          onBlur: () => {
            x(!1);
          },
          ref: v,
          ...g
        }
      ),
      /* @__PURE__ */ e("label", { className: "krds-form-chip-outline", htmlFor: N, children: r })
    ] });
  }
);
as.displayName = "RadioChip";
const ss = ({
  value: t,
  defaultValue: a,
  onChange: n,
  name: s,
  children: l,
  className: o,
  column: r = !1
}) => {
  const [c, d] = ne({
    value: t,
    defaultValue: a,
    onChange: n
  }), u = {
    value: c,
    onChange: d,
    name: s
  }, g = m("krds-check-area", r && "chk-column", o);
  return /* @__PURE__ */ e(xt.Provider, { value: u, children: /* @__PURE__ */ e("div", { className: g, children: l }) });
};
ss.displayName = "RadioGroup";
const rs = {
  sm: "작게",
  md: "보통",
  lg: "조금 크게",
  xlg: "크게",
  xxlg: "가장크게"
}, ls = ({ scale: t, label: a, isActive: n, onClick: s }) => {
  const l = () => {
    s(t);
  };
  return /* @__PURE__ */ e(
    $e.Item,
    {
      className: m(t, n && "active"),
      "data-adjust-scale": t,
      isActive: n,
      onClick: l,
      children: a
    }
  );
}, os = D(
  ({
    value: t,
    defaultValue: a = "md",
    onChange: n,
    buttonText: s = "화면크기",
    resetText: l = "초기화",
    labels: o = rs,
    className: r,
    ...c
  }, d) => {
    const [u, g] = ne({
      value: t,
      defaultValue: a,
      finalValue: "md",
      onChange: n
    }), v = (N) => {
      g(N);
      const L = document.querySelector(":root");
      if (L) {
        const O = getComputedStyle(L);
        let R = "";
        switch (N) {
          case "sm":
            R = O.getPropertyValue("--krds-zoom-small").trim();
            break;
          case "md":
            R = O.getPropertyValue("--krds-zoom-medium").trim();
            break;
          case "lg":
            R = O.getPropertyValue("--krds-zoom-large").trim();
            break;
          case "xlg":
            R = O.getPropertyValue("--krds-zoom-xlarge").trim();
            break;
          case "xxlg":
            R = O.getPropertyValue("--krds-zoom-xxlarge").trim();
            break;
        }
        R && (document.body.style.zoom = R);
      }
      n?.(N);
    }, y = () => {
      v("md");
    }, x = ["sm", "md", "lg", "xlg", "xxlg"], k = /* @__PURE__ */ h(Ee, { children: [
      s,
      " ",
      /* @__PURE__ */ e("i", { className: "svg-icon ico-toggle" })
    ] }), E = /* @__PURE__ */ h(
      "button",
      {
        type: "button",
        className: "krds-btn medium text",
        "data-adjust-scale": "md",
        onClick: y,
        children: [
          /* @__PURE__ */ e("i", { className: "svg-icon ico-reset" }),
          " ",
          l
        ]
      }
    );
    return /* @__PURE__ */ h(
      $e,
      {
        buttonText: k,
        className: m("krds-resize", r),
        ref: d,
        ...c,
        children: [
          /* @__PURE__ */ e($e.List, { children: x.map((N) => /* @__PURE__ */ e(
            ls,
            {
              scale: N,
              label: o[N],
              isActive: u === N,
              onClick: v
            },
            N
          )) }),
          /* @__PURE__ */ e($e.BottomSection, { children: E })
        ]
      }
    );
  }
);
os.displayName = "Resize";
const is = D(
  ({
    value: t,
    defaultValue: a,
    onChange: n,
    options: s,
    label: l,
    hint: o,
    error: r,
    size: c = "large",
    variant: d = "default",
    disabled: u = !1,
    className: g = "",
    ...v
  }, y) => {
    const [x, k] = ne({
      value: t,
      defaultValue: a,
      onChange: n
    }), E = m(
      d === "sorting" ? "krds-form-select-sort" : "krds-form-select",
      c,
      r && "is-error",
      u && "disabled",
      g
    );
    return d === "sorting" ? /* @__PURE__ */ e(
      "select",
      {
        className: E,
        disabled: u,
        value: x,
        onChange: (N) => k(N.target.value),
        title: "선택",
        ...v,
        ref: y,
        children: s.map((N) => /* @__PURE__ */ e("option", { value: N.value, children: N.label }, N.value))
      }
    ) : /* @__PURE__ */ h("div", { className: "form-group", children: [
      l && /* @__PURE__ */ e("div", { className: "form-tit", children: /* @__PURE__ */ e("label", { htmlFor: v.id, children: l }) }),
      /* @__PURE__ */ e("div", { className: "form-conts", children: /* @__PURE__ */ e(
        "select",
        {
          className: E,
          disabled: u,
          value: x,
          onChange: (N) => k(N.target.value),
          ...v,
          ref: y,
          children: s.map((N) => /* @__PURE__ */ e("option", { value: N.value, children: N.label }, N.value))
        }
      ) }),
      o && !r && /* @__PURE__ */ e("p", { className: "form-hint", children: o }),
      r && /* @__PURE__ */ e("p", { className: "form-hint-invalid", children: r })
    ] });
  }
);
is.displayName = "Select";
const Sn = D(
  ({ children: t, className: a, ...n }, s) => /* @__PURE__ */ e("nav", { className: m("krds-side-navigation", a), ref: s, ...n, children: t })
);
Sn.displayName = "SideNavigation";
const xn = D(
  ({ children: t, className: a, ...n }, s) => /* @__PURE__ */ e("h2", { className: m("lnb-tit", a), ref: s, ...n, children: t })
);
xn.displayName = "SideNavigation.Title";
const Dn = D(
  ({ children: t, className: a, ...n }, s) => /* @__PURE__ */ e("ul", { className: m("lnb-list", a), role: "menubar", ref: s, ...n, children: t })
);
Dn.displayName = "SideNavigation.Menu";
const Tn = D(
  ({ children: t, active: a, className: n, ...s }, l) => /* @__PURE__ */ e(
    "li",
    {
      className: m("lnb-item", a && "active", n),
      role: "none",
      ref: l,
      ...s,
      children: t
    }
  )
);
Tn.displayName = "SideNavigation.Item";
const En = D(
  ({ children: t, active: a, className: n, ...s }, l) => /* @__PURE__ */ e(
    "li",
    {
      className: m("lnb-subitem", a && "active", n),
      role: "none",
      ref: l,
      ...s,
      children: t
    }
  )
);
En.displayName = "SideNavigation.SubItem";
const Mn = D(
  ({ children: t, active: a, expanded: n, onClick: s, className: l, ...o }, r) => /* @__PURE__ */ e(
    "button",
    {
      type: "button",
      className: m("lnb-btn", "lnb-toggle", a && "active", l),
      role: "menuitem",
      "aria-expanded": n,
      onClick: s,
      ref: r,
      ...o,
      children: t
    }
  )
);
Mn.displayName = "SideNavigation.Toggle";
const Ln = D(
  ({ children: t, id: a, className: n, ...s }, l) => /* @__PURE__ */ e("div", { className: m("lnb-submenu", n), ref: l, ...s, children: /* @__PURE__ */ e("ul", { id: a, role: "menu", children: t }) })
);
Ln.displayName = "SideNavigation.SubMenu";
const In = D(
  ({ children: t, href: a = "#", current: n, onClick: s, className: l, ...o }, r) => /* @__PURE__ */ e(
    "a",
    {
      href: a,
      className: m("lnb-btn", "lnb-link", l),
      role: "menuitem",
      "aria-current": n ? "page" : void 0,
      onClick: s,
      ref: r,
      ...o,
      children: t
    }
  )
);
In.displayName = "SideNavigation.Link";
const Pn = D(
  ({ children: t, expanded: a, onClick: n, className: s, ...l }, o) => {
    const r = j(null), c = it(o, r), d = (u) => {
      const g = u.currentTarget, v = g.nextElementSibling;
      if (v && v.classList.contains("lnb-submenu-lv2")) {
        v.classList.add("active"), g.setAttribute("aria-expanded", "true");
        const y = () => {
          v.querySelector(".lnb-btn-tit")?.focus(), v.removeEventListener("transitionend", y);
        };
        v.addEventListener("transitionend", y);
      }
      n?.(u);
    };
    return /* @__PURE__ */ e(
      "button",
      {
        type: "button",
        className: m("lnb-btn", "lnb-toggle-popup", s),
        role: "menuitem",
        "aria-expanded": a,
        "aria-haspopup": "true",
        onClick: d,
        ref: c,
        ...l,
        children: t
      }
    );
  }
);
Pn.displayName = "SideNavigation.PopupToggle";
const An = D(
  ({ children: t, active: a, id: n, className: s, ...l }, o) => {
    const r = j(null), c = it(o, r);
    return J(() => {
      const d = r.current;
      if (!d) return;
      const u = (g) => {
        const v = g.relatedTarget;
        if (!d.contains(v)) {
          d.classList.remove("active");
          const y = d.previousElementSibling;
          if (y && y.classList.contains("lnb-toggle-popup")) {
            y.setAttribute("aria-expanded", "false");
            const x = () => {
              y.focus(), d.removeEventListener("transitionend", x);
            };
            d.addEventListener("transitionend", x);
          }
        }
      };
      return d.addEventListener("focusout", u), () => {
        d.removeEventListener("focusout", u);
      };
    }, []), /* @__PURE__ */ e(
      "div",
      {
        className: m("lnb-submenu-lv2", a && "active", s),
        id: n,
        role: "menu",
        ref: c,
        ...l,
        children: t
      }
    );
  }
);
An.displayName = "SideNavigation.Popup";
const Bn = D(
  ({ children: t, onClick: a, className: n, ...s }, l) => {
    const o = (r) => {
      const d = r.currentTarget.closest(".lnb-submenu-lv2");
      if (d) {
        d.classList.remove("active");
        const u = d.previousElementSibling;
        u && u.classList.contains("lnb-toggle-popup") && (u.setAttribute("aria-expanded", "false"), u.focus());
      }
      a?.(r);
    };
    return /* @__PURE__ */ e(
      "button",
      {
        type: "button",
        className: m("lnb-btn-tit", n),
        onClick: o,
        ref: l,
        ...s,
        children: t
      }
    );
  }
);
Bn.displayName = "SideNavigation.PopupTitle";
const $n = D(
  ({ children: t, className: a, ...n }, s) => /* @__PURE__ */ e("ul", { className: a, ref: s, ...n, children: t })
);
$n.displayName = "SideNavigation.PopupMenu";
const Ss = Object.assign(Sn, {
  Title: xn,
  Menu: Dn,
  Item: Tn,
  SubItem: En,
  Toggle: Mn,
  SubMenu: Ln,
  Link: In,
  PopupToggle: Pn,
  Popup: An,
  PopupTitle: Bn,
  PopupMenu: $n
}), cs = Re.forwardRef(
  ({ targetId: t = "breadcrumb", children: a = "본문 바로가기", className: n, ...s }, l) => /* @__PURE__ */ e("div", { id: "krds-skip-link", ref: l, className: n, ...s, children: /* @__PURE__ */ e("a", { href: `#${t}`, children: a }) })
);
cs.displayName = "SkipLink";
const ds = D(
  ({ steps: t, currentStep: a = -1, currentStepText: n = "현재단계", pageTitle: s, className: l, ...o }, r) => {
    const c = (u) => u < a ? "done" : u === a ? "active" : "default", d = /* @__PURE__ */ e("ol", { ref: r, className: m("krds-step-wrap", l), ...o, children: t.map((u, g) => {
      const v = c(g), y = v === "active", x = u.id || `step-${g}`;
      return /* @__PURE__ */ e("li", { className: v !== "default" ? v : void 0, children: /* @__PURE__ */ h("span", { children: [
        y && /* @__PURE__ */ e("em", { className: "sr-only", children: n }),
        /* @__PURE__ */ e("i", { className: "step", children: u.step }),
        /* @__PURE__ */ e("span", { className: "step-tit", children: u.title })
      ] }) }, x);
    }) });
    return s ? /* @__PURE__ */ h("div", { className: "page-title-wrap between", children: [
      /* @__PURE__ */ e("h2", { className: "h-tit", children: s }),
      d
    ] }) : d;
  }
);
ds.displayName = "StepIndicator";
const us = D(
  ({ children: t, className: a, ...n }, s) => {
    const l = m("krds-structured-list", "type-full", a);
    return /* @__PURE__ */ e("ul", { ref: s, className: l, ...n, children: t });
  }
);
us.displayName = "StructuredList";
const hs = D(
  ({ children: t, className: a, ...n }, s) => {
    const l = m("structured-item", a);
    return /* @__PURE__ */ e("li", { ref: s, className: l, ...n, children: /* @__PURE__ */ e("div", { className: "in", children: t }) });
  }
);
hs.displayName = "StructuredListItem";
const Rn = Ve(null), Dt = () => {
  const t = Oe(Rn);
  if (!t)
    throw new Error("Tab compound components must be used within a Tab component");
  return t;
}, On = D(
  ({
    value: t,
    defaultValue: a,
    onValueChange: n,
    variant: s = "line",
    size: l = "full",
    children: o,
    className: r,
    ...c
  }, d) => {
    const u = j(/* @__PURE__ */ new Map()), g = j([]), [v, y] = ne({
      value: t,
      defaultValue: a,
      onChange: n,
      finalValue: ""
    }), x = F((E, N) => (E ? (u.current.has(N) || g.current.push(N), u.current.set(N, E)) : (u.current.delete(N), g.current = g.current.filter((L) => L !== N)), g.current.indexOf(N)), []), k = {
      activeValue: v,
      onValueChange: y,
      variant: s,
      size: l,
      registerTrigger: x,
      triggerOrder: g,
      triggerRefs: u
    };
    return /* @__PURE__ */ e(Rn.Provider, { value: k, children: /* @__PURE__ */ e("div", { ref: d, className: m("krds-tab-area", r), ...c, children: o }) });
  }
), Hn = D(
  ({ children: t, className: a, ...n }, s) => {
    const { variant: l, size: o } = Dt(), r = m("tab", l, o === "full" && "full");
    return /* @__PURE__ */ e("div", { className: r, children: /* @__PURE__ */ e("ul", { ref: s, role: "tablist", className: a, ...n, children: t }) });
  }
), vt = D(
  ({ value: t, children: a, className: n, disabled: s, ...l }, o) => {
    const { activeValue: r, onValueChange: c, registerTrigger: d, triggerOrder: u, triggerRefs: g } = Dt(), v = j(null), [y, x] = ee(-1), k = r === t, E = F(
      (M) => {
        v.current = M;
        const S = d(M, t);
        x(S), typeof o == "function" ? o(M) : o && (o.current = M);
      },
      [t, d, o]
    ), N = F(() => {
      s || c?.(t);
    }, [t, c, s]), L = F(
      (M) => {
        if (y === -1) return;
        const S = u.current?.length || 0;
        let b = null;
        switch (M.key) {
          case "ArrowRight":
            M.preventDefault(), b = y + 1 >= S ? 0 : y + 1;
            break;
          case "ArrowLeft":
            M.preventDefault(), b = y - 1 < 0 ? S - 1 : y - 1;
            break;
          case "Home":
            M.preventDefault(), b = 0;
            break;
          case "End":
            M.preventDefault(), b = S - 1;
            break;
        }
        if (b !== null && b !== y) {
          const T = u.current?.[b] || "";
          g.current?.get(T)?.focus();
        }
      },
      [y, u, g]
    ), O = `tab_${t}`, R = `panel_${t}`;
    return /* @__PURE__ */ e("li", { role: "presentation", className: m("tab-item", k && "active"), children: /* @__PURE__ */ h(
      "button",
      {
        role: "tab",
        id: O,
        ref: E,
        "aria-selected": k,
        "aria-controls": R,
        type: "button",
        className: m("btn-tab", n),
        onClick: N,
        onKeyDown: L,
        disabled: s,
        ...l,
        children: [
          a,
          k && /* @__PURE__ */ e("i", { className: "sr-only", children: "선택됨" })
        ]
      }
    ) });
  }
), ms = D(
  ({ children: t, className: a, ...n }, s) => /* @__PURE__ */ e("div", { ref: s, className: m("tab-conts-wrap", a), ...n, children: t })
), Vn = D(
  ({ value: t, children: a, className: n, ...s }, l) => {
    const { activeValue: o } = Dt(), r = o === t, c = `tab_${t}`, d = `panel_${t}`, u = m("tab-conts", r && "active", n);
    return /* @__PURE__ */ e(
      "section",
      {
        ref: l,
        id: d,
        role: "tabpanel",
        "aria-labelledby": c,
        className: u,
        ...s,
        children: a
      }
    );
  }
);
On.displayName = "Tab";
Hn.displayName = "TabList";
vt.displayName = "TabTrigger";
Vn.displayName = "TabPanel";
const ps = D(
  ({ children: t, className: a, ...n }, s) => {
    const l = m("krds-btn-tag", n.variant === "link" && "link", a);
    if (n.variant === "link") {
      const { variant: u, ...g } = n;
      return /* @__PURE__ */ e("a", { ref: s, className: l, ...g, children: t });
    }
    const { variant: o, onDelete: r, deleteDisabled: c, ...d } = n;
    return /* @__PURE__ */ h("span", { ref: s, className: l, ...d, children: [
      t,
      r && /* @__PURE__ */ e("button", { type: "button", className: "btn-delete", onClick: r, disabled: c, children: /* @__PURE__ */ e("span", { className: "sr-only", children: "삭제" }) })
    ] });
  }
);
ps.displayName = "Tag";
const fs = D(
  ({ children: t, size: a = "medium", className: n, ...s }, l) => {
    const o = m("krds-tag-wrap", a, n);
    return /* @__PURE__ */ e("div", { ref: l, className: o, ...s, children: t });
  }
);
fs.displayName = "TagWrap";
const gs = D(
  ({
    value: t,
    defaultValue: a,
    onChange: n,
    label: s,
    maxLength: l,
    showCount: o = !0,
    countTotal: r = 100,
    disabled: c = !1,
    readOnly: d = !1,
    className: u = "",
    id: g,
    ...v
  }, y) => {
    const x = Me(), k = g || x, [E, N] = ne({
      value: t,
      defaultValue: a,
      onChange: n
    }), L = le(() => E ? E.length : 0, [E]), O = o || !!l, R = m("krds-input", u), M = m("form-group");
    return /* @__PURE__ */ h("div", { className: M, children: [
      s && /* @__PURE__ */ e("div", { className: "form-tit", children: /* @__PURE__ */ e("label", { htmlFor: k, children: s }) }),
      /* @__PURE__ */ e("div", { className: "form-conts", children: /* @__PURE__ */ h("div", { className: "textarea-wrap", children: [
        /* @__PURE__ */ e(
          "textarea",
          {
            ref: y,
            id: k,
            className: R,
            value: E,
            onChange: (S) => N(S.target.value),
            disabled: c,
            readOnly: d,
            maxLength: l,
            ...v
          }
        ),
        O && /* @__PURE__ */ h("p", { className: "textarea-count", children: [
          /* @__PURE__ */ e("span", { className: "count-now", children: L }),
          /* @__PURE__ */ h("span", { className: "count-total", children: [
            "/",
            l ?? r
          ] })
        ] })
      ] }) })
    ] });
  }
);
gs.displayName = "Textarea";
const bs = D(
  ({
    value: t,
    defaultValue: a,
    onChange: n,
    label: s,
    hint: l,
    error: o,
    success: r,
    information: c,
    size: d = "large",
    showPasswordToggle: u = !1,
    showClearButton: g = !1,
    type: v = "text",
    disabled: y = !1,
    readOnly: x = !1,
    className: k = "",
    id: E,
    ...N
  }, L) => {
    const O = Me(), R = E || O, [M, S] = ne({
      value: t,
      defaultValue: a,
      onChange: n,
      finalValue: ""
    }), [b, T] = ee(!1), [B, Y] = ee(v);
    J(() => {
      Y(v === "password" ? b ? "text" : "password" : v);
    }, [v, b]);
    const V = !!o, U = !!r && !V, G = !!c && !V && !U, K = u && v === "password", H = g && M && !y && !x, _ = K || H, te = m(
      "form-group",
      V && "is-error",
      U && "is-success",
      G && "is-information",
      k
    ), ie = m("form-conts", _ && "btn-ico-wrap"), ce = () => {
      S("");
    }, oe = () => {
      T((Q) => !Q);
    }, $ = () => V ? /* @__PURE__ */ e("p", { className: "form-hint-invalid", children: o }) : U ? /* @__PURE__ */ e("p", { className: "form-hint-success", children: r }) : G ? /* @__PURE__ */ e("p", { className: "form-hint-information", children: c }) : l ? /* @__PURE__ */ e("p", { className: "form-hint", children: l }) : null;
    return /* @__PURE__ */ h("div", { className: te, children: [
      s && /* @__PURE__ */ e("div", { className: "form-tit", children: /* @__PURE__ */ e("label", { htmlFor: R, children: s }) }),
      /* @__PURE__ */ h(
        "div",
        {
          className: ie,
          ...H && { "data-delete": "true" },
          children: [
            /* @__PURE__ */ e(
              "input",
              {
                ref: L,
                id: R,
                type: B,
                className: m("krds-input", d),
                value: M,
                onChange: (Q) => S(Q.target.value),
                disabled: y,
                readOnly: x,
                ...N
              }
            ),
            _ && /* @__PURE__ */ h("div", { className: "btn-group", children: [
              H && /* @__PURE__ */ h(
                re,
                {
                  variant: "icon",
                  size: "medium",
                  className: "pure btn-delete-input",
                  onClick: ce,
                  children: [
                    /* @__PURE__ */ e("span", { className: "sr-only", children: "내용 삭제" }),
                    /* @__PURE__ */ e("i", { className: "svg-icon ico-delete-fill" })
                  ]
                }
              ),
              K && /* @__PURE__ */ h(re, { variant: "icon", size: "medium", onClick: oe, children: [
                /* @__PURE__ */ e("span", { className: "sr-only", children: b ? "입력한 비밀번호 가리기" : "입력한 비밀번호 보기" }),
                /* @__PURE__ */ e(
                  "i",
                  {
                    className: m(
                      "svg-icon ico-pw-visible",
                      b && "ico-pw-visible-on",
                      !b && "ico-pw-visible"
                    )
                  }
                )
              ] })
            ] })
          ]
        }
      ),
      $()
    ] });
  }
);
bs.displayName = "TextInput";
const vs = D(
  ({
    size: t = "large",
    label: a,
    checked: n,
    defaultChecked: s = !1,
    onChange: l,
    className: o,
    id: r,
    disabled: c,
    ...d
  }, u) => {
    const g = Me(), v = r || g, [y, x] = ee(!1), [k, E] = ne({
      value: n,
      defaultValue: s,
      finalValue: !1,
      onChange: (M, S) => {
        c || l?.(M, S);
      }
    }), N = (M) => {
      c || E(M.target.checked, M);
    }, L = () => {
      x(!0);
    }, O = () => {
      x(!1);
    }, R = m(
      "krds-form-toggle-switch",
      t === "medium" && "medium",
      t === "large" && "large",
      y && "focus",
      o
    );
    return /* @__PURE__ */ h("div", { className: R, children: [
      /* @__PURE__ */ e(
        "input",
        {
          ref: u,
          type: "checkbox",
          id: v,
          checked: k,
          disabled: c,
          onChange: N,
          onFocus: L,
          onBlur: O,
          ...d
        }
      ),
      /* @__PURE__ */ h("label", { htmlFor: v, children: [
        /* @__PURE__ */ e("span", { className: "switch-toggle", children: /* @__PURE__ */ e("i", {}) }),
        a
      ] })
    ] });
  }
);
vs.displayName = "ToggleSwitch";
const Ns = D(
  ({ children: t, text: a, variant: n = "vertical", className: s }, l) => {
    const [o, r] = ee(!1), [c, d] = ee({
      top: 0,
      left: 0,
      isTop: !1,
      isBottom: !1,
      isLeft: !1,
      isRight: !1,
      isMobileSmall: !1
    }), u = j(null), g = j(null), v = `tooltip-popover-${Me()}`, y = Re.useCallback(() => {
      if (!u.current || !g.current) return;
      const M = 12, { clientHeight: S, clientWidth: b } = g.current, {
        top: T,
        left: B,
        right: Y,
        height: V,
        width: U
      } = u.current.getBoundingClientRect(), G = window.innerWidth / 2, K = window.innerHeight / 2;
      let H, _, te = !1, ie = !1, ce = !1, oe = !1;
      window.innerWidth <= 768 || n === "box" || n === "vertical" ? (T + V > K ? (H = T - S - M, te = !0) : (H = T + V + M, ie = !0), B + U > G ? (_ = Y - b, oe = !0, window.innerWidth - (B + U) > b / 2 && (_ = B + (U - b) / 2, oe = !1)) : (_ = B + (U - b) / 2, _ < 0 && (_ = B, ce = !0))) : (H = T + (V - S) / 2, B + U > G ? (_ = B - b - M, oe = !0) : _ = Y + M);
      const me = window.innerWidth <= 420;
      me && (_ = window.innerWidth / 2), d({
        top: H,
        left: _,
        isTop: te,
        isBottom: ie,
        isLeft: ce,
        isRight: oe,
        isMobileSmall: me
      });
    }, [n]), x = () => {
      r(!0);
    }, k = () => {
      r(!1), d({
        top: 0,
        left: 0,
        isTop: !1,
        isBottom: !1,
        isLeft: !1,
        isRight: !1,
        isMobileSmall: !1
      });
    };
    J(() => {
      o && y();
    }, [o, y]), J(() => {
      const M = (T) => {
        (T.key === "Escape" || T.key === "Esc") && k();
      }, S = () => {
        k();
      }, b = () => {
        k();
      };
      return document.addEventListener("keydown", M), window.addEventListener("scroll", S), window.addEventListener("resize", b), () => {
        document.removeEventListener("keydown", M), window.removeEventListener("scroll", S), window.removeEventListener("resize", b);
      };
    }, []);
    const E = (M) => {
      if (x(), Ke(t)) {
        const S = t;
        S.props.onMouseEnter && S.props.onMouseEnter(M);
      }
    }, N = (M) => {
      if (k(), Ke(t)) {
        const S = t;
        S.props.onMouseLeave && S.props.onMouseLeave(M);
      }
    }, L = (M) => {
      if (x(), Ke(t)) {
        const S = t;
        S.props.onFocus && S.props.onFocus(M);
      }
    }, O = (M) => {
      if (k(), Ke(t)) {
        const S = t;
        S.props.onBlur && S.props.onBlur(M);
      }
    }, R = (M) => {
      u && (u.current = M), typeof l == "function" ? l(M) : l && (l.current = M);
    };
    if (Ke(t)) {
      const M = t;
      return /* @__PURE__ */ h(Ee, { children: [
        yt(M, {
          // @ts-expect-error - valid ref
          ref: R,
          onMouseEnter: E,
          onMouseLeave: N,
          onFocus: L,
          onBlur: O,
          className: m("krds-tooltip", s, M.props.className),
          "data-tooltip": a,
          "aria-labelledby": v
        }),
        /* @__PURE__ */ e(
          "div",
          {
            ref: g,
            id: v,
            className: m(
              "krds-tooltip-popover",
              n === "box" && "tooltip-box",
              n === "vertical" && "tooltip-vertical",
              o && "active",
              c.isTop && "top",
              c.isBottom && "bottom",
              c.isLeft && "left",
              c.isRight && "right"
            ),
            style: {
              ...o && {
                top: `${c.top}px`,
                left: c.isMobileSmall ? "50%" : `${c.left}px`,
                transform: c.isMobileSmall ? "translateX(-50%)" : void 0
              }
            },
            "aria-hidden": !o,
            children: a
          }
        )
      ] });
    }
    return /* @__PURE__ */ h(Ee, { children: [
      /* @__PURE__ */ e(
        "button",
        {
          ref: R,
          onMouseEnter: E,
          onMouseLeave: N,
          onFocus: L,
          onBlur: O,
          className: m("krds-tooltip", s),
          "data-tooltip": a,
          "aria-labelledby": v,
          children: t
        }
      ),
      /* @__PURE__ */ e(
        "div",
        {
          ref: g,
          id: v,
          className: m(
            "krds-tooltip-popover",
            n === "box" && "tooltip-box",
            n === "vertical" && "tooltip-vertical",
            o && "active",
            c.isTop && "top",
            c.isBottom && "bottom",
            c.isLeft && "left",
            c.isRight && "right"
          ),
          style: {
            ...o && {
              top: `${c.top}px`,
              left: c.isMobileSmall ? "50%" : `${c.left}px`,
              transform: c.isMobileSmall ? "translateX(-50%)" : void 0
            }
          },
          "aria-hidden": !o,
          children: a
        }
      )
    ] });
  }
);
Ns.displayName = "Tooltip";
const Fn = Ve(null), pt = () => {
  const t = Oe(Fn);
  if (!t)
    throw new Error("TutorialPanel 컴포넌트들은 TutorialPanel.Root 내부에서 사용해야 합니다.");
  return t;
}, Nt = D(
  ({
    isOpen: t,
    onOpenChange: a,
    activeTab: n,
    defaultActiveTab: s,
    onTabChange: l,
    children: o,
    className: r,
    ...c
  }, d) => {
    const [u, g] = ne({
      value: t,
      defaultValue: !1,
      onChange: a,
      finalValue: !1
    }), [v, y] = ne({
      value: n,
      defaultValue: s,
      onChange: l,
      finalValue: "help"
    }), x = j(null), k = j(null), E = j(null), N = j(null), L = j(null), O = F(() => window.innerWidth <= 1023 ? "mob" : "pc", []), R = F(() => {
      const T = document.querySelector("#krds-masthead")?.clientHeight || 0, B = document.querySelector("#krds-header .header-in")?.clientHeight || 0, Y = T + B, V = B, U = k.current, G = x.current, K = E.current;
      if (!U || !G || !K) return;
      const H = (_) => {
        U.style.marginTop = _, O() === "pc" ? (G.style.paddingTop = _, K.style.marginTop = _) : (G.removeAttribute("style"), K.removeAttribute("style"));
      };
      document.body.classList.contains("bn-hidden") ? document.querySelector("#wrap")?.classList.contains("scroll-down") ? H("0") : H(`${V}px`) : H(`${Y}px`);
    }, [O]), M = F(() => {
      const T = document.querySelector("#krds-masthead");
      if (!T) return;
      const B = document.body;
      L.current = new IntersectionObserver(
        (Y) => {
          Y.forEach((V) => {
            B.classList.toggle("bn-hidden", !V.isIntersecting), R();
          });
        },
        { root: null, threshold: 0 }
      ), L.current.observe(T);
    }, [R]), S = F(
      (T) => {
        const B = x.current, Y = document.querySelector("#container > .inner");
        T ? (O() === "mob" && document.body.classList.add("scroll-no"), B && (B.setAttribute("tabindex", "0"), setTimeout(() => {
          B.focus();
        }, 50)), document.querySelectorAll(".btn-help-exec").forEach((U) => {
          U.setAttribute("aria-expanded", "true");
        }), Y?.classList.contains("help-panel-flexible") && Y.classList.add("help-panel-expanded")) : (O() === "mob" && document.body.classList.remove("scroll-no"), B && B.removeAttribute("tabindex"), N.current ? N.current.focus() : k.current?.focus(), document.querySelectorAll(".btn-help-exec").forEach((U) => {
          U.setAttribute("aria-expanded", "false");
        }), Y?.classList.contains("help-panel-flexible") && Y.classList.remove("help-panel-expanded")), g(T);
      },
      [O, g]
    ), b = {
      isOpen: u,
      onOpenChange: S,
      activeTab: v,
      onTabChange: y
    };
    return J(() => {
      R(), M();
      const T = () => {
        R(), O() === "mob" && u ? document.body.classList.add("scroll-no") : document.body.classList.remove("scroll-no");
      };
      return window.addEventListener("resize", T), k.current && k.current.setAttribute("aria-expanded", u ? "true" : "false"), () => {
        window.removeEventListener("resize", T), L.current && L.current.disconnect(), document.body.classList.remove("scroll-no");
      };
    }, [R, M, u, O]), /* @__PURE__ */ e(Fn.Provider, { value: b, children: /* @__PURE__ */ e(
      "div",
      {
        id: "container",
        ref: d,
        className: m("inner help-panel-flexible", r),
        ...c,
        children: o
      }
    ) });
  }
), _n = D(
  ({ children: t = "도움말", className: a, onClick: n, ...s }, l) => {
    const { isOpen: o, onOpenChange: r } = pt(), c = (u) => {
      r(!o), n?.(u);
    }, d = (u) => /* @__PURE__ */ e("i", { className: `svg-icon ${u}` });
    return /* @__PURE__ */ h(
      re,
      {
        ref: l,
        size: "small",
        variant: "tertiary",
        className: m("btn-help-panel expand btn-help-exec", a),
        onClick: c,
        "aria-expanded": o,
        "aria-label": typeof t == "string" ? t : "도움말",
        ...s,
        children: [
          d("ico-fold"),
          " ",
          t
        ]
      }
    );
  }
), zn = D(
  ({ children: t, className: a, ...n }, s) => {
    const { isOpen: l } = pt();
    return /* @__PURE__ */ e(
      "div",
      {
        ref: s,
        className: m("krds-help-panel", l && "expand", a),
        ...n,
        children: /* @__PURE__ */ e("div", { className: "help-panel-wrap", children: /* @__PURE__ */ e("div", { className: "help-conts-area", children: t }) })
      }
    );
  }
), Yn = D(
  ({ children: t, className: a, id: n, role: s }, l) => {
    const { activeTab: o, onTabChange: r } = pt();
    return /* @__PURE__ */ h(
      On,
      {
        ref: l,
        value: o,
        onValueChange: (c) => r(c),
        variant: "line",
        size: "normal",
        className: m("layer", a),
        id: n,
        role: s,
        children: [
          /* @__PURE__ */ h(Hn, { children: [
            /* @__PURE__ */ e(vt, { value: "help", children: "도움" }),
            /* @__PURE__ */ e(vt, { value: "tutorial", children: "따라하기" })
          ] }),
          /* @__PURE__ */ e(ms, { children: t })
        ]
      }
    );
  }
), Kn = D(
  ({ value: t, children: a, className: n, ...s }, l) => /* @__PURE__ */ h(Vn, { ref: l, value: t, className: n, ...s, children: [
    /* @__PURE__ */ e("h3", { className: "sr-only", children: t === "help" ? "도움" : "따라하기" }),
    a
  ] })
), qn = D(
  ({ helpContent: t, relatedServices: a = [], className: n, ...s }, l) => {
    const o = (r) => /* @__PURE__ */ e("i", { className: `svg-icon ${r}` });
    return /* @__PURE__ */ h("div", { ref: l, className: m("help-conts-area-inner", n), ...s, children: [
      t && /* @__PURE__ */ e("div", { className: "conts-area help-conts", children: /* @__PURE__ */ h("div", { className: "conts-wrap", children: [
        /* @__PURE__ */ h("h4", { className: "help-title", children: [
          t.title,
          /* @__PURE__ */ h("span", { className: "krds-btn medium icon", children: [
            /* @__PURE__ */ e("span", { className: "sr-only", children: "도움말" }),
            o("ico-help")
          ] })
        ] }),
        t.description && /* @__PURE__ */ e("div", { className: "conts-desc", children: t.description }),
        t.links && t.links.length > 0 && /* @__PURE__ */ e("ul", { className: "link-list", children: t.links.map((r, c) => /* @__PURE__ */ e("li", { children: /* @__PURE__ */ h(
          re,
          {
            as: "a",
            href: r.href,
            target: r.external ? "_blank" : void 0,
            title: r.external ? "새 창 열림" : void 0,
            size: "xsmall",
            variant: "link",
            className: "basic",
            children: [
              r.icon && r.iconPosition === "left" && o(r.icon),
              r.label,
              r.icon && r.iconPosition !== "left" && o(r.icon),
              !r.icon && (r.external ? o("ico-go") : o("ico-angle right"))
            ]
          }
        ) }, c)) })
      ] }) }),
      a.length > 0 && /* @__PURE__ */ e("div", { className: "conts-area related-service", children: a.map((r, c) => /* @__PURE__ */ h("div", { className: "conts-wrap", children: [
        /* @__PURE__ */ e("h4", { className: "help-title", children: r.title }),
        r.description && /* @__PURE__ */ e("div", { className: "conts-desc", children: r.description }),
        r.links && r.links.length > 0 && /* @__PURE__ */ e("ul", { className: "link-list", children: r.links.map((d, u) => /* @__PURE__ */ e("li", { children: /* @__PURE__ */ h(
          re,
          {
            as: "a",
            href: d.href,
            target: d.external ? "_blank" : void 0,
            title: d.external ? "새 창 열림" : void 0,
            size: "xsmall",
            variant: "link",
            className: "basic",
            children: [
              d.icon && d.iconPosition === "left" && o(d.icon),
              d.label,
              d.icon && d.iconPosition !== "left" && o(d.icon),
              !d.icon && (d.external ? o("ico-go") : o("ico-angle right"))
            ]
          }
        ) }, u)) })
      ] }, c)) })
    ] });
  }
), Wn = D(
  ({ title: t, steps: a = [], onTutorialStop: n, stopButtonText: s = "그만 따라하기", className: l, ...o }, r) => /* @__PURE__ */ h("div", { ref: r, className: m("help-conts-area-inner", l), ...o, children: [
    /* @__PURE__ */ h("div", { className: "conts-area", children: [
      t && /* @__PURE__ */ e("h4", { className: "help-title", children: /* @__PURE__ */ e("a", { href: "#", title: "이전으로 돌아가기", children: t }) }),
      a.length > 0 && /* @__PURE__ */ e("ul", { className: "coach-help-process", children: a.map((c, d) => /* @__PURE__ */ h("li", { children: [
        /* @__PURE__ */ e("h4", { className: m("tit", c.current && "current"), children: c.title }),
        c.steps && c.steps.length > 0 && /* @__PURE__ */ e(
          kt,
          {
            buttonText: `전체 ${c.steps.length}단계`,
            className: "conts-expand-area",
            children: /* @__PURE__ */ e("ul", { className: "krds-info-list decimal", children: c.steps.map((u, g) => /* @__PURE__ */ e("li", { children: u }, g)) })
          }
        ),
        c.content && /* @__PURE__ */ e("div", { className: "step-content", children: c.content })
      ] }, d)) })
    ] }),
    /* @__PURE__ */ e("div", { className: "help-panel-action", children: /* @__PURE__ */ e(
      re,
      {
        size: "medium",
        variant: "secondary",
        className: "coach-btn-stop",
        onClick: n,
        children: s
      }
    ) })
  ] })
), Un = D(
  ({ children: t, className: a, onClick: n, ...s }, l) => {
    const { onOpenChange: o } = pt(), r = (d) => {
      o(!1), n?.(d);
    }, c = (d) => /* @__PURE__ */ e("i", { className: `svg-icon ${d}` });
    return /* @__PURE__ */ e(
      re,
      {
        ref: l,
        size: "small",
        variant: "tertiary",
        className: m("btn-help-panel fold", a),
        onClick: r,
        ...s,
        children: t || /* @__PURE__ */ h(Ee, { children: [
          /* @__PURE__ */ e("span", { className: "sr-only", children: "도움말" }),
          " 접어두기 ",
          c("ico-fold right")
        ] })
      }
    );
  }
);
Nt.displayName = "TutorialPanel.Root";
_n.displayName = "TutorialPanel.Trigger";
zn.displayName = "TutorialPanel.Container";
Yn.displayName = "TutorialPanel.Tabs";
Kn.displayName = "TutorialPanel.TabPanel";
qn.displayName = "TutorialPanel.HelpContent";
Wn.displayName = "TutorialPanel.TutorialContent";
Un.displayName = "TutorialPanel.Close";
const xs = Object.assign(Nt, {
  Root: Nt,
  Trigger: _n,
  Container: zn,
  Tabs: Yn,
  TabPanel: Kn,
  HelpContent: qn,
  TutorialContent: Wn,
  Close: Un
});
export {
  Qn as Accordion,
  Rt as AccordionItem,
  Zn as Badge,
  ta as Breadcrumb,
  re as Button,
  aa as Calendar,
  sa as Checkbox,
  ra as CheckboxChip,
  la as CheckboxGroup,
  oa as CoachMark,
  ia as ContextualHelp,
  Cs as CriticalAlert,
  da as DateInput,
  kt as Disclosure,
  ha as FileUpload,
  Na as Footer,
  qe as Header,
  Ia as HelpPanel,
  _t as Identifier,
  Ya as InPageNavigation,
  pn as LanguageSwitcher,
  Xa as Link,
  Ja as Masthead,
  es as Modal,
  ts as Pagination,
  ns as Radio,
  as as RadioChip,
  ss as RadioGroup,
  os as Resize,
  is as Select,
  Ss as SideNavigation,
  cs as SkipLink,
  ua as Spinner,
  ds as StepIndicator,
  us as StructuredList,
  On as Tab,
  ms as TabContent,
  Hn as TabList,
  Vn as TabPanel,
  vt as TabTrigger,
  ps as Tag,
  bs as TextInput,
  on as TextList,
  gs as Textarea,
  vs as ToggleSwitch,
  Ns as Tooltip,
  xs as TutorialPanel
};
